// primeng/components/rating/rating.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/rating/rating");
var i2 = require("@angular/common");
var i3 = require("@angular/forms");
exports.RatingModuleNgFactory = i0.ɵcmf(i1.RatingModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.RatingModule, i1.RatingModule, [])]);
});
var styles_Rating = [];
exports.RenderType_Rating = i0.ɵcrt({ encapsulation: 2, styles: styles_Rating,
    data: {} });
function View_Rating_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'a', [['href', '#']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.clear($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-ban']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_Rating_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'a', [['href', '#']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.rate($event, _v.context.index) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class', 'fa']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        i0.ɵpod({ 'fa-star-o': 0, 'fa-star': 1 }), (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'fa';
        var currVal_1 = _ck(_v, 4, 0, (!_co.value || (_v.context.index >= _co.value)), (_v.context.index < _co.value));
        _ck(_v, 3, 0, currVal_0, currVal_1);
    }, null);
}
function View_Rating_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 9, 'div', [['class', 'ui-rating']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Rating_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Rating_2)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-rating';
        var currVal_1 = _ck(_v, 3, 0, _co.disabled);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.cancel;
        _ck(_v, 6, 0, currVal_2);
        var currVal_3 = _co.starsArray;
        _ck(_v, 9, 0, currVal_3);
    }, null);
}
exports.View_Rating_0 = View_Rating_0;
function View_Rating_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-rating', [], null, null, null, View_Rating_0, exports.RenderType_Rating)),
        i0.ɵprd(5120, null, i3.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i1.Rating]), i0.ɵdid(114688, null, 0, i1.Rating, [], null, null)], function (_ck, _v) {
        _ck(_v, 2, 0);
    }, null);
}
exports.View_Rating_Host_0 = View_Rating_Host_0;
exports.RatingNgFactory = i0.ɵccf('p-rating', i1.Rating, View_Rating_Host_0, { disabled: 'disabled', readonly: 'readonly', stars: 'stars', cancel: 'cancel' }, { onRate: 'onRate', onCancel: 'onCancel' }, []);
