// primeng/components/breadcrumb/breadcrumb.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/breadcrumb/breadcrumb");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
exports.BreadcrumbModuleNgFactory = i0.ɵcmf(i1.BreadcrumbModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.RouterModule, i3.RouterModule, [[2, i3.ɵa], [2, i3.Router]]), i0.ɵmpd(512, i1.BreadcrumbModule, i1.BreadcrumbModule, [])]);
});
var styles_Breadcrumb = [];
exports.RenderType_Breadcrumb = i0.ɵcrt({ encapsulation: 2, styles: styles_Breadcrumb,
    data: {} });
function View_Breadcrumb_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'li', [['class',
                'ui-breadcrumb-home fa fa-home']], null, null, null, null, null))], null, null);
}
function View_Breadcrumb_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'a', [['class',
                'ui-menuitem-link']], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.itemClick($event, _co.home) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-home']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_3 = 'ui-menuitem-link';
        var currVal_4 = _ck(_v, 2, 0, _co.home.disabled);
        _ck(_v, 1, 0, currVal_3, currVal_4);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.home.url || '#');
        var currVal_1 = _co.home.target;
        var currVal_2 = _co.home.title;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
    });
}
function View_Breadcrumb_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 10, 'a', [['class',
                'ui-menuitem-link']], [[1, 'target', 0], [1, 'title', 0], [1, 'target', 0], [8, 'href',
                4]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 3).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.itemClick($event, _co.home) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), i0.ɵdid(671744, [[2,
                4]], 0, i3.RouterLinkWithHref, [i3.Router, i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null), i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef, i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'], routerLinkActive: [1, 'routerLinkActive'] }, null), i0.ɵqud(603979776, 1, { links: 1 }), i0.ɵqud(603979776, 2, { linksWithHrefs: 1 }),
        i0.ɵpod({ exact: 0 }), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-home']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_4 = 'ui-menuitem-link';
        var currVal_5 = _ck(_v, 2, 0, _co.home.disabled);
        _ck(_v, 1, 0, currVal_4, currVal_5);
        var currVal_6 = _co.home.routerLink;
        _ck(_v, 3, 0, currVal_6);
        var currVal_7 = (_co.home.routerLinkActiveOptions || _ck(_v, 7, 0, false));
        var currVal_8 = 'ui-state-active';
        _ck(_v, 4, 0, currVal_7, currVal_8);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.home.target;
        var currVal_1 = _co.home.title;
        var currVal_2 = i0.ɵnov(_v, 3).target;
        var currVal_3 = i0.ɵnov(_v, 3).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
    });
}
function View_Breadcrumb_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 7, 'li', [['class',
                'ui-breadcrumb-home']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_4)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = !_co.home.routerLink;
        _ck(_v, 3, 0, currVal_0);
        var currVal_1 = _co.home.routerLink;
        _ck(_v, 6, 0, currVal_1);
    }, null);
}
function View_Breadcrumb_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'li', [['class',
                'ui-breadcrumb-chevron fa fa-chevron-right']], null, null, null, null, null))], null, null);
}
function View_Breadcrumb_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 6, 'a', [['class',
                'ui-menuitem-link']], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.itemClick($event, _v.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), (_l()(), i0.ɵted(null, ['\n                            '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(),
            i0.ɵted(null, ['\n                        ']))], function (_ck, _v) {
        var currVal_3 = 'ui-menuitem-link';
        var currVal_4 = _ck(_v, 2, 0, _v.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_3, currVal_4);
    }, function (_ck, _v) {
        var currVal_0 = (_v.parent.context.$implicit.url || '#');
        var currVal_1 = _v.parent.context.$implicit.target;
        var currVal_2 = _v.parent.context.$implicit.title;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_5 = _v.parent.context.$implicit.label;
        _ck(_v, 5, 0, currVal_5);
    });
}
function View_Breadcrumb_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 11, 'a', [['class',
                'ui-menuitem-link']], [[1, 'target', 0], [1, 'title', 0], [1, 'target', 0], [8, 'href',
                4]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 3).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.itemClick($event, _v.parent.context.$implicit) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), i0.ɵdid(671744, [[4,
                4]], 0, i3.RouterLinkWithHref, [i3.Router, i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null), i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef, i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'], routerLinkActive: [1, 'routerLinkActive'] }, null), i0.ɵqud(603979776, 3, { links: 1 }), i0.ɵqud(603979776, 4, { linksWithHrefs: 1 }),
        i0.ɵpod({ exact: 0 }), (_l()(), i0.ɵted(null, ['\n                            '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                        ']))], function (_ck, _v) {
        var currVal_4 = 'ui-menuitem-link';
        var currVal_5 = _ck(_v, 2, 0, _v.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_4, currVal_5);
        var currVal_6 = _v.parent.context.$implicit.routerLink;
        _ck(_v, 3, 0, currVal_6);
        var currVal_7 = (_v.parent.context.$implicit.routerLinkActiveOptions || _ck(_v, 7, 0, false));
        var currVal_8 = 'ui-state-active';
        _ck(_v, 4, 0, currVal_7, currVal_8);
    }, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.target;
        var currVal_1 = _v.parent.context.$implicit.title;
        var currVal_2 = i0.ɵnov(_v, 3).target;
        var currVal_3 = i0.ɵnov(_v, 3).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
        var currVal_9 = _v.parent.context.$implicit.label;
        _ck(_v, 10, 0, currVal_9);
    });
}
function View_Breadcrumb_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'li', [['class',
                'ui-breadcrumb-chevron fa fa-chevron-right']], null, null, null, null, null))], null, null);
}
function View_Breadcrumb_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                    '])), (_l()(),
            i0.ɵeld(0, null, null, 7, 'li', [['role', 'menuitem']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_7)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_8)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_9)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var currVal_0 = !_v.context.$implicit.routerLink;
        _ck(_v, 4, 0, currVal_0);
        var currVal_1 = _v.context.$implicit.routerLink;
        _ck(_v, 7, 0, currVal_1);
        var currVal_2 = !_v.context.last;
        _ck(_v, 11, 0, currVal_2);
    }, null);
}
function View_Breadcrumb_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 18, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 13, 'ul', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_5)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Breadcrumb_6)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-breadcrumb ui-widget ui-widget-header ui-helper-clearfix ui-corner-all';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = !_co.home;
        _ck(_v, 8, 0, currVal_3);
        var currVal_4 = _co.home;
        _ck(_v, 11, 0, currVal_4);
        var currVal_5 = _co.model;
        _ck(_v, 14, 0, currVal_5);
        var currVal_6 = _co.model;
        _ck(_v, 17, 0, currVal_6);
    }, null);
}
exports.View_Breadcrumb_0 = View_Breadcrumb_0;
function View_Breadcrumb_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-breadcrumb', [], null, null, null, View_Breadcrumb_0, exports.RenderType_Breadcrumb)),
        i0.ɵdid(49152, null, 0, i1.Breadcrumb, [], null, null)], null, null);
}
exports.View_Breadcrumb_Host_0 = View_Breadcrumb_Host_0;
exports.BreadcrumbNgFactory = i0.ɵccf('p-breadcrumb', i1.Breadcrumb, View_Breadcrumb_Host_0, { model: 'model', style: 'style', styleClass: 'styleClass',
    home: 'home' }, {}, []);
