// primeng/components/overlaypanel/overlaypanel.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/overlaypanel/overlaypanel");
var i2 = require("@angular/common");
var i3 = require("primeng/components/dom/domhandler");
exports.OverlayPanelModuleNgFactory = i0.ɵcmf(i1.OverlayPanelModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.OverlayPanelModule, i1.OverlayPanelModule, [])]);
});
var styles_OverlayPanel = [];
exports.RenderType_OverlayPanel = i0.ɵcrt({ encapsulation: 2, styles: styles_OverlayPanel,
    data: {} });
function View_OverlayPanel_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'a', [['class',
                'ui-overlaypanel-close ui-state-default'], ['href', '#']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onCloseClick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-fw fa-close']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_OverlayPanel_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 11, 'div', [], [[4, 'display', null]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onPanelClick() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-overlaypanel-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OverlayPanel_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _co.styleClass;
        var currVal_2 = 'ui-overlaypanel ui-widget ui-widget-content ui-corner-all ui-shadow';
        _ck(_v, 2, 0, currVal_1, currVal_2);
        var currVal_3 = _co.style;
        _ck(_v, 3, 0, currVal_3);
        var currVal_4 = _co.showCloseIcon;
        _ck(_v, 11, 0, currVal_4);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.visible ? 'block' : 'none');
        _ck(_v, 1, 0, currVal_0);
    });
}
exports.View_OverlayPanel_0 = View_OverlayPanel_0;
function View_OverlayPanel_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-overlayPanel', [], null, null, null, View_OverlayPanel_0, exports.RenderType_OverlayPanel)), i0.ɵprd(512, null, i3.DomHandler, i3.DomHandler, []), i0.ɵdid(4440064, null, 0, i1.OverlayPanel, [i0.ElementRef,
            i3.DomHandler, i0.Renderer2, i0.ChangeDetectorRef], null, null)], function (_ck, _v) {
        _ck(_v, 2, 0);
    }, null);
}
exports.View_OverlayPanel_Host_0 = View_OverlayPanel_Host_0;
exports.OverlayPanelNgFactory = i0.ɵccf('p-overlayPanel', i1.OverlayPanel, View_OverlayPanel_Host_0, { dismissable: 'dismissable', showCloseIcon: 'showCloseIcon',
    style: 'style', styleClass: 'styleClass', appendTo: 'appendTo' }, { onBeforeShow: 'onBeforeShow',
    onAfterShow: 'onAfterShow', onBeforeHide: 'onBeforeHide', onAfterHide: 'onAfterHide' }, ['*']);
