// primeng/components/dropdown/dropdown.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/dropdown/dropdown");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
var i4 = require("@angular/forms");
var i5 = require("primeng/components/dom/domhandler");
var i6 = require("primeng/components/utils/objectutils");
exports.DropdownModuleNgFactory = i0.ɵcmf(i1.DropdownModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i1.DropdownModule, i1.DropdownModule, [])]);
});
var styles_Dropdown = [];
exports.RenderType_Dropdown = i0.ɵcrt({ encapsulation: 2, styles: styles_Dropdown,
    data: { 'animation': [{ type: 7, name: 'panelState', definitions: [{ type: 0, name: 'hidden',
                        styles: { type: 6, styles: { opacity: 0 }, offset: null }, options: undefined },
                    { type: 0, name: 'visible', styles: { type: 6, styles: { opacity: 1 }, offset: null },
                        options: undefined }, { type: 1, expr: 'visible => hidden', animation: { type: 4,
                            styles: null, timings: '400ms ease-in' }, options: null }, { type: 1,
                        expr: 'hidden => visible', animation: { type: 4, styles: null, timings: '400ms ease-out' },
                        options: null }], options: {} }] } });
function View_Dropdown_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'option', [], [[8, 'value', 0], [8, 'selected', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit.value;
        var currVal_1 = (_co.selectedOption == _v.context.$implicit);
        _ck(_v, 0, 0, currVal_0, currVal_1);
        var currVal_2 = _v.context.$implicit.label;
        _ck(_v, 1, 0, currVal_2);
    });
}
function View_Dropdown_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 7, 'div', [['class',
                'ui-helper-hidden-accessible']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵeld(0, null, null, 4, 'select', [['aria-hidden', 'true'], ['name',
                    'options'], ['tabindex', '-1']], [[8, 'required', 0], [1, 'aria-label', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_2)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = _co.options;
        _ck(_v, 5, 0, currVal_2);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.required;
        var currVal_1 = (_co.selectedOption ? _co.selectedOption.label : ' ');
        _ck(_v, 2, 0, currVal_0, currVal_1);
    });
}
function View_Dropdown_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'label', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngClass: [0, 'ngClass'] }, null), i0.ɵpod({ 'ui-dropdown-label ui-inputtext ui-corner-all': 0,
            'ui-dropdown-label-empty': 1 }), (_l()(), i0.ɵted(null, ['', '']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _ck(_v, 2, 0, true, (_co.label === null));
        _ck(_v, 1, 0, currVal_0);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = (_co.label || 'empty');
        _ck(_v, 3, 0, currVal_1);
    });
}
function View_Dropdown_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'label', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngClass: [0, 'ngClass'] }, null), i0.ɵpod({ 'ui-dropdown-label ui-inputtext ui-corner-all ui-placeholder': 0 }),
        (_l()(), i0.ɵted(null, ['', '']))], function (_ck, _v) {
        var currVal_0 = _ck(_v, 2, 0, true);
        _ck(_v, 1, 0, currVal_0);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _co.placeholder;
        _ck(_v, 3, 0, currVal_1);
    });
}
function View_Dropdown_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, [[6, 0], ['editableInput', 1]], null, 0, 'input', [['class', 'ui-dropdown-label ui-inputtext ui-corner-all'], ['type', 'text']], [[1, 'aria-label', 0], [8, 'disabled', 0], [1, 'placeholder', 0]], [[null, 'click'],
            [null, 'input'], [null, 'focus'], [null, 'blur']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onEditableInputClick($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('input' === en)) {
                var pd_1 = (_co.onEditableInputChange($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('focus' === en)) {
                var pd_2 = (_co.onEditableInputFocus($event) !== false);
                ad = (pd_2 && ad);
            }
            if (('blur' === en)) {
                var pd_3 = (_co.onInputBlur($event) !== false);
                ad = (pd_3 && ad);
            }
            return ad;
        }, null, null))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.selectedOption ? _co.selectedOption.label : ' ');
        var currVal_1 = _co.disabled;
        var currVal_2 = _co.placeholder;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
    });
}
function View_Dropdown_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'div', [['class',
                'ui-dropdown-filter-container']], null, [[null, 'input'], [null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('input' === en)) {
                var pd_0 = (_co.onFilter($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = ($event.stopPropagation() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵeld(0, [[4, 0], ['filter', 1]], null, 0, 'input', [['autocomplete',
                'off'], ['class', 'ui-dropdown-filter ui-inputtext ui-widget ui-state-default ui-corner-all'],
            ['type', 'text']], [[1, 'placeholder', 0]], [[null, 'keydown.enter'],
            [null, 'keydown']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('keydown.enter' === en)) {
                var pd_0 = ($event.preventDefault() !== false);
                ad = (pd_0 && ad);
            }
            if (('keydown' === en)) {
                var pd_1 = (_co.onKeydown($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-search']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.filterPlaceholder;
        _ck(_v, 2, 0, currVal_0);
    });
}
function View_Dropdown_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var currVal_0 = (_v.parent.context.$implicit.label || 'empty');
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_Dropdown_11(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_Dropdown_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_11)), i0.ɵdid(212992, null, 0, i3.TemplateWrapper, [i0.ViewContainerRef], { item: [0, 'item'], templateRef: [1, 'templateRef'] }, null), (_l()(), i0.ɵand(0, null, null, 0))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.parent.context.$implicit;
        var currVal_1 = _co.itemTemplate;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Dropdown_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 9, 'li', [], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onItemClick($event, _v.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-dropdown-item ui-corner-all': 0, 'ui-state-highlight': 1, 'ui-dropdown-item-empty': 2 }),
        (_l()(), i0.ɵted(null, ['\n                            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_9)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_10)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _ck(_v, 2, 0, true, (_co.selectedOption == _v.context.$implicit), (!_v.context.$implicit.label || (_v.context.$implicit.label.length === 0)));
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = !_co.itemTemplate;
        _ck(_v, 5, 0, currVal_1);
        var currVal_2 = _co.itemTemplate;
        _ck(_v, 8, 0, currVal_2);
    }, null);
}
function View_Dropdown_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'ul', [['class',
                'ui-dropdown-items ui-dropdown-list ui-widget-content ui-widget ui-corner-all ui-helper-reset']], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_8)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.optionsToDisplay;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_Dropdown_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { containerViewChild: 0 }), i0.ɵqud(402653184, 2, { panelViewChild: 0 }), i0.ɵqud(402653184, 3, { itemsWrapperViewChild: 0 }), i0.ɵqud(671088640, 4, { filterViewChild: 0 }), i0.ɵqud(402653184, 5, { focusViewChild: 0 }), i0.ɵqud(671088640, 6, { editableInputViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n         '])),
        (_l()(), i0.ɵeld(0, [[1, 0], ['container', 1]], null, 40, 'div', [], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onMouseclick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-dropdown ui-widget ui-state-default ui-corner-all ui-helper-clearfix': 0,
            'ui-state-disabled': 1, 'ui-dropdown-open': 2, 'ui-state-focus': 3 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'ui-helper-hidden-accessible']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, [[5,
                0], ['in', 1]], null, 0, 'input', [['readonly', ''], ['role', 'listbox'],
            ['type', 'text']], [[1, 'id', 0], [1, 'aria-label', 0], [8, 'disabled', 0], [1, 'tabindex',
                0], [1, 'autofocus', 0]], [[null, 'focus'], [null, 'blur'], [null,
                'keydown']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('focus' === en)) {
                var pd_0 = (_co.onInputFocus($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('blur' === en)) {
                var pd_1 = (_co.onInputBlur($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('keydown' === en)) {
                var pd_2 = (_co.onKeydown($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Dropdown_4)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_5)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'ui-dropdown-trigger ui-state-default ui-corner-right']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-fw fa-caret-down ui-clickable']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, [[2, 0], ['panel', 1]], null, 12, 'div', [], [[24, '@panelState', 0], [4, 'display', null]], null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_6)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, [[3, 0],
            ['itemswrapper', 1]], null, 4, 'div', [['class', 'ui-dropdown-items-wrapper']], [[4, 'max-height', null]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dropdown_7)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 9, 0, true, _co.disabled, _co.panelVisible, _co.focus);
        _ck(_v, 8, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 10, 0, currVal_2);
        var currVal_3 = _co.autoWidth;
        _ck(_v, 13, 0, currVal_3);
        var currVal_9 = (!_co.editable && (_co.label !== null));
        _ck(_v, 21, 0, currVal_9);
        var currVal_10 = (!_co.editable && (_co.label === null));
        _ck(_v, 24, 0, currVal_10);
        var currVal_11 = _co.editable;
        _ck(_v, 27, 0, currVal_11);
        var currVal_14 = _co.panelStyleClass;
        var currVal_15 = 'ui-dropdown-panel ui-widget-content ui-corner-all ui-helper-hidden ui-shadow';
        _ck(_v, 35, 0, currVal_14, currVal_15);
        var currVal_16 = _co.panelStyle;
        _ck(_v, 36, 0, currVal_16);
        var currVal_17 = _co.filter;
        _ck(_v, 39, 0, currVal_17);
        var currVal_19 = (_co.lazy ? _co.panelVisible : true);
        _ck(_v, 44, 0, currVal_19);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_4 = _co.inputId;
        var currVal_5 = (_co.selectedOption ? _co.selectedOption.label : ' ');
        var currVal_6 = _co.disabled;
        var currVal_7 = _co.tabindex;
        var currVal_8 = _co.autofocus;
        _ck(_v, 17, 0, currVal_4, currVal_5, currVal_6, currVal_7, currVal_8);
        var currVal_12 = (_co.panelVisible ? 'visible' : 'hidden');
        var currVal_13 = (_co.panelVisible ? 'block' : 'none');
        _ck(_v, 34, 0, currVal_12, currVal_13);
        var currVal_18 = (_co.scrollHeight || 'auto');
        _ck(_v, 41, 0, currVal_18);
    });
}
exports.View_Dropdown_0 = View_Dropdown_0;
function View_Dropdown_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'p-dropdown', [], null, null, null, View_Dropdown_0, exports.RenderType_Dropdown)),
        i0.ɵprd(5120, null, i4.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i1.Dropdown]), i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []),
        i0.ɵprd(512, null, i6.ObjectUtils, i6.ObjectUtils, []), i0.ɵdid(13877248, null, 1, i1.Dropdown, [i0.ElementRef, i5.DomHandler, i0.Renderer2, i0.ChangeDetectorRef,
            i6.ObjectUtils, i0.NgZone], null, null), i0.ɵqud(603979776, 1, { templates: 1 })], function (_ck, _v) {
        _ck(_v, 4, 0);
    }, null);
}
exports.View_Dropdown_Host_0 = View_Dropdown_Host_0;
exports.DropdownNgFactory = i0.ɵccf('p-dropdown', i1.Dropdown, View_Dropdown_Host_0, { scrollHeight: 'scrollHeight', filter: 'filter',
    style: 'style', panelStyle: 'panelStyle', styleClass: 'styleClass', panelStyleClass: 'panelStyleClass',
    disabled: 'disabled', readonly: 'readonly', autoWidth: 'autoWidth', required: 'required',
    editable: 'editable', appendTo: 'appendTo', tabindex: 'tabindex', placeholder: 'placeholder',
    filterPlaceholder: 'filterPlaceholder', inputId: 'inputId', dataKey: 'dataKey',
    filterBy: 'filterBy', lazy: 'lazy', autofocus: 'autofocus', options: 'options' }, { onChange: 'onChange',
    onFocus: 'onFocus', onBlur: 'onBlur' }, []);
