// primeng/components/tieredmenu/tieredmenu.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/tieredmenu/tieredmenu");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
var i4 = require("primeng/components/dom/domhandler");
exports.TieredMenuModuleNgFactory = i0.ɵcmf(i1.TieredMenuModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.RouterModule, i3.RouterModule, [[2, i3.ɵa], [2, i3.Router]]), i0.ɵmpd(512, i1.TieredMenuModule, i1.TieredMenuModule, [])]);
});
var styles_TieredMenu = [];
exports.RenderType_TieredMenu = i0.ɵcrt({ encapsulation: 2, styles: styles_TieredMenu,
    data: {} });
function View_TieredMenu_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 8, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-tieredmenu ui-menu ui-widget ui-widget-content ui-corner-all ui-helper-clearfix': 0,
            'ui-menu-dynamic ui-shadow': 1 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 2, 'p-tieredMenuSub', [['root', 'root']], null, null, null, View_TieredMenuSub_0, exports.RenderType_TieredMenuSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.TieredMenuSub, [i4.DomHandler], { item: [0, 'item'], root: [1, 'root'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 3, 0, true, _co.popup);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 4, 0, currVal_2);
        var currVal_3 = _co.model;
        var currVal_4 = 'root';
        _ck(_v, 8, 0, currVal_3, currVal_4);
    }, null);
}
exports.View_TieredMenu_0 = View_TieredMenu_0;
function View_TieredMenu_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-tieredMenu', [], null, null, null, View_TieredMenu_0, exports.RenderType_TieredMenu)),
        i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(4374528, null, 0, i1.TieredMenu, [i0.ElementRef, i4.DomHandler, i0.Renderer2], null, null)], null, null);
}
exports.View_TieredMenu_Host_0 = View_TieredMenu_Host_0;
exports.TieredMenuNgFactory = i0.ɵccf('p-tieredMenu', i1.TieredMenu, View_TieredMenu_Host_0, { model: 'model', popup: 'popup', style: 'style',
    styleClass: 'styleClass', appendTo: 'appendTo' }, {}, []);
var styles_TieredMenuSub = [];
exports.RenderType_TieredMenuSub = i0.ɵcrt({ encapsulation: 2,
    styles: styles_TieredMenuSub, data: {} });
function View_TieredMenuSub_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'li', [['class',
                'ui-menu-separator ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                ']))], null, null);
}
function View_TieredMenuSub_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class',
                'ui-submenu-icon fa fa-fw fa-caret-right']], null, null, null, null, null))], null, null);
}
function View_TieredMenuSub_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _v.parent.parent.parent.context.$implicit.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_TieredMenuSub_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 12, 'a', [['class',
                'ui-menuitem-link ui-corner-all']], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.itemClick($event, _v.parent.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_5)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_6)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_3 = 'ui-menuitem-link ui-corner-all';
        var currVal_4 = _ck(_v, 2, 0, _v.parent.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_3, currVal_4);
        var currVal_5 = _v.parent.parent.context.$implicit.items;
        _ck(_v, 5, 0, currVal_5);
        var currVal_6 = _v.parent.parent.context.$implicit.icon;
        _ck(_v, 8, 0, currVal_6);
    }, function (_ck, _v) {
        var currVal_0 = (_v.parent.parent.context.$implicit.url || '#');
        var currVal_1 = _v.parent.parent.context.$implicit.target;
        var currVal_2 = _v.parent.parent.context.$implicit.title;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_7 = _v.parent.parent.context.$implicit.label;
        _ck(_v, 11, 0, currVal_7);
    });
}
function View_TieredMenuSub_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class',
                'ui-submenu-icon fa fa-fw fa-caret-right']], null, null, null, null, null))], null, null);
}
function View_TieredMenuSub_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _v.parent.parent.parent.context.$implicit.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_TieredMenuSub_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 17, 'a', [['class',
                'ui-menuitem-link ui-corner-all']], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0],
            [1, 'target', 0], [8, 'href', 4]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 3).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.itemClick($event, _v.parent.parent.context.$implicit) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), i0.ɵdid(671744, [[2,
                4]], 0, i3.RouterLinkWithHref, [i3.Router, i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null), i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef, i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'], routerLinkActive: [1, 'routerLinkActive'] }, null), i0.ɵqud(603979776, 1, { links: 1 }), i0.ɵqud(603979776, 2, { linksWithHrefs: 1 }),
        i0.ɵpod({ exact: 0 }), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_8)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_9)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_5 = 'ui-menuitem-link ui-corner-all';
        var currVal_6 = _ck(_v, 2, 0, _v.parent.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_5, currVal_6);
        var currVal_7 = _v.parent.parent.context.$implicit.routerLink;
        _ck(_v, 3, 0, currVal_7);
        var currVal_8 = (_v.parent.parent.context.$implicit.routerLinkActiveOptions || _ck(_v, 7, 0, false));
        var currVal_9 = 'ui-state-active';
        _ck(_v, 4, 0, currVal_8, currVal_9);
        var currVal_10 = _v.parent.parent.context.$implicit.items;
        _ck(_v, 10, 0, currVal_10);
        var currVal_11 = _v.parent.parent.context.$implicit.icon;
        _ck(_v, 13, 0, currVal_11);
    }, function (_ck, _v) {
        var currVal_0 = (_v.parent.parent.context.$implicit.url || '#');
        var currVal_1 = _v.parent.parent.context.$implicit.target;
        var currVal_2 = _v.parent.parent.context.$implicit.title;
        var currVal_3 = i0.ɵnov(_v, 3).target;
        var currVal_4 = i0.ɵnov(_v, 3).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4);
        var currVal_12 = _v.parent.parent.context.$implicit.label;
        _ck(_v, 16, 0, currVal_12);
    });
}
function View_TieredMenuSub_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-tieredMenuSub', [['class', 'ui-submenu']], null, null, null, View_TieredMenuSub_0, exports.RenderType_TieredMenuSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.TieredMenuSub, [i4.DomHandler], { item: [0, 'item'] }, null)], function (_ck, _v) {
        var currVal_0 = _v.parent.parent.context.$implicit;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
function View_TieredMenuSub_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, [['listItem', 1]], null, 12, 'li', [], null, [[null, 'mouseenter'], [null, 'mouseleave']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('mouseenter' === en)) {
                var pd_0 = (_co.onItemMouseEnter($event, i0.ɵnov(_v, 0), _v.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            if (('mouseleave' === en)) {
                var pd_1 = (_co.onItemMouseLeave($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-menuitem ui-widget ui-corner-all': 0, 'ui-menu-parent': 1, 'ui-menuitem-active': 2 }),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_4)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_7)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_10)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _ck(_v, 2, 0, true, _v.parent.context.$implicit.items, (i0.ɵnov(_v, 0) == _co.activeItem));
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = !_v.parent.context.$implicit.routerLink;
        _ck(_v, 5, 0, currVal_1);
        var currVal_2 = _v.parent.context.$implicit.routerLink;
        _ck(_v, 8, 0, currVal_2);
        var currVal_3 = _v.parent.context.$implicit.items;
        _ck(_v, 11, 0, currVal_3);
    }, null);
}
function View_TieredMenuSub_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.separator;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.separator;
        _ck(_v, 4, 0, currVal_1);
    }, null);
}
function View_TieredMenuSub_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'ul', [['class', 'ui-menu-list']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.listClick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-helper-reset': 0, 'ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-child ui-shadow': 1 }),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TieredMenuSub_1)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-menu-list';
        var currVal_1 = _ck(_v, 3, 0, _co.root, !_co.root);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = (_co.root ? _co.item : _co.item.items);
        _ck(_v, 6, 0, currVal_2);
    }, null);
}
exports.View_TieredMenuSub_0 = View_TieredMenuSub_0;
function View_TieredMenuSub_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-tieredMenuSub', [], null, null, null, View_TieredMenuSub_0, exports.RenderType_TieredMenuSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.TieredMenuSub, [i4.DomHandler], null, null)], null, null);
}
exports.View_TieredMenuSub_Host_0 = View_TieredMenuSub_Host_0;
exports.TieredMenuSubNgFactory = i0.ɵccf('p-tieredMenuSub', i1.TieredMenuSub, View_TieredMenuSub_Host_0, { item: 'item', root: 'root' }, {}, []);
