// primeng/components/menu/menu.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/menu/menu");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
var i4 = require("primeng/components/dom/domhandler");
exports.MenuModuleNgFactory = i0.ɵcmf(i1.MenuModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.RouterModule, i3.RouterModule, [[2, i3.ɵa], [2, i3.Router]]), i0.ɵmpd(512, i1.MenuModule, i1.MenuModule, [])]);
});
var styles_Menu = [];
exports.RenderType_Menu = i0.ɵcrt({ encapsulation: 2, styles: styles_Menu,
    data: {} });
function View_Menu_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'li', [['class',
                'ui-menu-separator ui-widget-content']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-helper-hidden': 0 })], function (_ck, _v) {
        var currVal_0 = 'ui-menu-separator ui-widget-content';
        var currVal_1 = _ck(_v, 2, 0, (_v.parent.context.$implicit.visible === false));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Menu_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'li', [['class',
                'ui-widget-header ui-corner-all']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-helper-hidden': 0 }), (_l()(), i0.ɵeld(0, null, null, 1, 'h3', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['',
            '']))], function (_ck, _v) {
        var currVal_0 = 'ui-widget-header ui-corner-all';
        var currVal_1 = _ck(_v, 2, 0, (_v.parent.context.$implicit.visible === false));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, function (_ck, _v) {
        var currVal_2 = _v.parent.context.$implicit.label;
        _ck(_v, 4, 0, currVal_2);
    });
}
function View_Menu_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'li', [['class',
                'ui-menu-separator ui-widget-content']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-helper-hidden': 0 })], function (_ck, _v) {
        var currVal_0 = 'ui-menu-separator ui-widget-content';
        var currVal_1 = _ck(_v, 2, 0, ((_v.parent.context.$implicit.visible === false) || (_v.parent.parent.context.$implicit.visible === false)));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Menu_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'li', [['class',
                'ui-menuitem ui-widget ui-corner-all']], null, null, null, View_MenuItemContent_0, exports.RenderType_MenuItemContent)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-helper-hidden': 0 }),
        i0.ɵdid(49152, null, 0, i1.MenuItemContent, [i1.Menu], { item: [0, 'item'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem ui-widget ui-corner-all';
        var currVal_1 = _ck(_v, 2, 0, ((_v.parent.context.$implicit.visible === false) || (_v.parent.parent.context.$implicit.visible === false)));
        _ck(_v, 1, 0, currVal_0, currVal_1);
        var currVal_2 = _v.parent.context.$implicit;
        _ck(_v, 3, 0, currVal_2);
    }, null);
}
function View_Menu_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Menu_6)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Menu_7)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.separator;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.separator;
        _ck(_v, 5, 0, currVal_1);
    }, null);
}
function View_Menu_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                    '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Menu_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Menu_4)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Menu_5)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.separator;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.separator;
        _ck(_v, 5, 0, currVal_1);
        var currVal_2 = _v.context.$implicit.items;
        _ck(_v, 8, 0, currVal_2);
    }, null);
}
function View_Menu_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(16777216, null, null, 1, null, View_Menu_2)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef,
            i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(),
            i0.ɵand(0, null, null, 0))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.model;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
function View_Menu_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'li', [['class',
                'ui-menu-separator ui-widget-content']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-helper-hidden': 0 })], function (_ck, _v) {
        var currVal_0 = 'ui-menu-separator ui-widget-content';
        var currVal_1 = _ck(_v, 2, 0, (_v.parent.context.$implicit.visible === false));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Menu_11(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'li', [['class',
                'ui-menuitem ui-widget ui-corner-all']], null, null, null, View_MenuItemContent_0, exports.RenderType_MenuItemContent)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-helper-hidden': 0 }),
        i0.ɵdid(49152, null, 0, i1.MenuItemContent, [i1.Menu], { item: [0, 'item'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem ui-widget ui-corner-all';
        var currVal_1 = _ck(_v, 2, 0, (_v.parent.context.$implicit.visible === false));
        _ck(_v, 1, 0, currVal_0, currVal_1);
        var currVal_2 = _v.parent.context.$implicit;
        _ck(_v, 3, 0, currVal_2);
    }, null);
}
function View_Menu_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                    '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Menu_10)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Menu_11)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.separator;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.separator;
        _ck(_v, 5, 0, currVal_1);
    }, null);
}
function View_Menu_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(16777216, null, null, 1, null, View_Menu_9)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef,
            i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(),
            i0.ɵand(0, null, null, 0))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.model;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
function View_Menu_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { containerViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, [[1, 0], ['container', 1]], null, 13, 'div', [], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = ((_co.preventDocumentDefault = true) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-menu ui-widget ui-widget-content ui-corner-all ui-helper-clearfix': 0,
            'ui-menu-dynamic ui-shadow': 1 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 7, 'ul', [['class',
                'ui-menu-list ui-helper-reset']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Menu_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Menu_8)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 4, 0, true, _co.popup);
        _ck(_v, 3, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 5, 0, currVal_2);
        var currVal_3 = _co.hasSubMenu();
        _ck(_v, 10, 0, currVal_3);
        var currVal_4 = !_co.hasSubMenu();
        _ck(_v, 13, 0, currVal_4);
    }, null);
}
exports.View_Menu_0 = View_Menu_0;
function View_Menu_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-menu', [], null, [['window', 'resize']], function (_v, en, $event) {
            var ad = true;
            if (('window:resize' === en)) {
                var pd_0 = (i0.ɵnov(_v, 2).onResize($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_Menu_0, exports.RenderType_Menu)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(4374528, null, 0, i1.Menu, [i0.ElementRef, i4.DomHandler,
            i0.Renderer2], null, null)], null, null);
}
exports.View_Menu_Host_0 = View_Menu_Host_0;
exports.MenuNgFactory = i0.ɵccf('p-menu', i1.Menu, View_Menu_Host_0, { model: 'model', popup: 'popup', style: 'style', styleClass: 'styleClass',
    appendTo: 'appendTo' }, {}, []);
var styles_MenuItemContent = [];
exports.RenderType_MenuItemContent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_MenuItemContent, data: {} });
function View_MenuItemContent_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _co.item.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_MenuItemContent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 9, 'a', [['class',
                'ui-menuitem-link ui-corner-all']], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.menu.itemClick($event, _co.item) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenuItemContent_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(),
            i0.ɵted(null, ['\n        ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_3 = 'ui-menuitem-link ui-corner-all';
        var currVal_4 = _ck(_v, 2, 0, _co.item.disabled);
        _ck(_v, 1, 0, currVal_3, currVal_4);
        var currVal_5 = _co.item.icon;
        _ck(_v, 5, 0, currVal_5);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.item.url || '#');
        var currVal_1 = _co.item.target;
        var currVal_2 = _co.item.title;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_6 = _co.item.label;
        _ck(_v, 8, 0, currVal_6);
    });
}
function View_MenuItemContent_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _co.item.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_MenuItemContent_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 14, 'a', [['class',
                'ui-menuitem-link ui-corner-all']], [[1, 'target', 0], [1, 'title', 0], [1, 'target',
                0], [8, 'href', 4]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 3).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.menu.itemClick($event, _co.item) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), i0.ɵdid(671744, [[2,
                4]], 0, i3.RouterLinkWithHref, [i3.Router, i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null), i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef, i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'], routerLinkActive: [1, 'routerLinkActive'] }, null), i0.ɵqud(603979776, 1, { links: 1 }), i0.ɵqud(603979776, 2, { linksWithHrefs: 1 }),
        i0.ɵpod({ exact: 0 }), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_MenuItemContent_4)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n        ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_4 = 'ui-menuitem-link ui-corner-all';
        var currVal_5 = _ck(_v, 2, 0, _co.item.disabled);
        _ck(_v, 1, 0, currVal_4, currVal_5);
        var currVal_6 = _co.item.routerLink;
        _ck(_v, 3, 0, currVal_6);
        var currVal_7 = (_co.item.routerLinkActiveOptions || _ck(_v, 7, 0, false));
        var currVal_8 = 'ui-state-active';
        _ck(_v, 4, 0, currVal_7, currVal_8);
        var currVal_9 = _co.item.icon;
        _ck(_v, 10, 0, currVal_9);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.item.target;
        var currVal_1 = _co.item.title;
        var currVal_2 = i0.ɵnov(_v, 3).target;
        var currVal_3 = i0.ɵnov(_v, 3).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
        var currVal_10 = _co.item.label;
        _ck(_v, 13, 0, currVal_10);
    });
}
function View_MenuItemContent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenuItemContent_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenuItemContent_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = !_co.item.routerLink;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = _co.item.routerLink;
        _ck(_v, 5, 0, currVal_1);
    }, null);
}
exports.View_MenuItemContent_0 = View_MenuItemContent_0;
function View_MenuItemContent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['pMenuItemContent',
                '']], null, null, null, View_MenuItemContent_0, exports.RenderType_MenuItemContent)),
        i0.ɵdid(49152, null, 0, i1.MenuItemContent, [i1.Menu], { item: [0, 'item'] }, null)], function (_ck, _v) {
        var currVal_0 = '';
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
exports.View_MenuItemContent_Host_0 = View_MenuItemContent_Host_0;
exports.MenuItemContentNgFactory = i0.ɵccf('[pMenuItemContent]', i1.MenuItemContent, View_MenuItemContent_Host_0, { item: 'pMenuItemContent' }, {}, []);
