// primeng/components/captcha/captcha.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/captcha/captcha");
var i2 = require("@angular/common");
exports.CaptchaModuleNgFactory = i0.ɵcmf(i1.CaptchaModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.CaptchaModule, i1.CaptchaModule, [])]);
});
var styles_Captcha = [];
exports.RenderType_Captcha = i0.ɵcrt({ encapsulation: 2, styles: styles_Captcha,
    data: {} });
function View_Captcha_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'div', [], null, null, null, null, null))], null, null);
}
exports.View_Captcha_0 = View_Captcha_0;
function View_Captcha_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-captcha', [], null, null, null, View_Captcha_0, exports.RenderType_Captcha)),
        i0.ɵdid(4374528, null, 0, i1.Captcha, [i0.ElementRef, i0.NgZone], null, null)], null, null);
}
exports.View_Captcha_Host_0 = View_Captcha_Host_0;
exports.CaptchaNgFactory = i0.ɵccf('p-captcha', i1.Captcha, View_Captcha_Host_0, { siteKey: 'siteKey', theme: 'theme', type: 'type', size: 'size',
    tabindex: 'tabindex', language: 'language', initCallback: 'initCallback' }, { onResponse: 'onResponse',
    onExpire: 'onExpire' }, []);
