// primeng/components/dialog/dialog.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/dialog/dialog");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
var i4 = require("primeng/components/dom/domhandler");
exports.DialogModuleNgFactory = i0.ɵcmf(i1.DialogModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i1.DialogModule, i1.DialogModule, [])]);
});
var styles_Dialog = [];
exports.RenderType_Dialog = i0.ɵcrt({ encapsulation: 2, styles: styles_Dialog,
    data: { 'animation': [{ type: 7, name: 'dialogState', definitions: [{ type: 0, name: 'hidden',
                        styles: { type: 6, styles: { opacity: 0 }, offset: null }, options: undefined },
                    { type: 0, name: 'visible', styles: { type: 6, styles: { opacity: 1 }, offset: null },
                        options: undefined }, { type: 1, expr: 'visible => hidden', animation: { type: 4,
                            styles: null, timings: '400ms ease-in' }, options: null }, { type: 1,
                        expr: 'hidden => visible', animation: { type: 4, styles: null, timings: '400ms ease-out' },
                        options: null }], options: {} }] } });
function View_Dialog_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-dialog-title']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.header;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_Dialog_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'span', [['class',
                'ui-dialog-title']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n                ']))], null, null);
}
function View_Dialog_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'a', [['href', '#'],
            ['role', 'button']], null, [[null, 'click'], [null, 'mousedown']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.close($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('mousedown' === en)) {
                var pd_1 = (_co.onCloseMouseDown($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-dialog-titlebar-icon ui-dialog-titlebar-close ui-corner-all': 0 }),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-fw fa-close']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var currVal_0 = _ck(_v, 2, 0, true);
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
function View_Dialog_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, [[2, 0], ['titlebar', 1]], null, 10, 'div', [['class', 'ui-dialog-titlebar ui-widget-header ui-helper-clearfix ui-corner-top']], null, [[null, 'mousedown'], [null, 'mouseup']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('mousedown' === en)) {
                var pd_0 = (_co.initDrag($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('mouseup' === en)) {
                var pd_1 = (_co.endDrag($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dialog_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dialog_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dialog_4)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.header;
        _ck(_v, 3, 0, currVal_0);
        var currVal_1 = _co.headerFacet;
        _ck(_v, 6, 0, currVal_1);
        var currVal_2 = _co.closable;
        _ck(_v, 9, 0, currVal_2);
    }, null);
}
function View_Dialog_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-dialog-footer ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 2), (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_Dialog_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'div', [['class',
                'ui-resizable-handle ui-resizable-se ui-icon ui-icon-gripsmall-diagonal-se'],
            ['style', 'z-index: 90;']], null, [[null, 'mousedown']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('mousedown' === en)) {
                var pd_0 = (_co.initResize($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null))], null, null);
}
function View_Dialog_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { containerViewChild: 0 }), i0.ɵqud(671088640, 2, { headerViewChild: 0 }), i0.ɵqud(402653184, 3, { contentViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, [[1, 0], ['container', 1]], null, 19, 'div', [], [[4, 'display', null], [4, 'width', 'px'], [4, 'height',
                'px'], [4, 'minWidth', 'px'], [24, '@dialogState', 0]], [[null, 'mousedown']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('mousedown' === en)) {
                var pd_0 = (_co.moveOnTop() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-dialog ui-widget ui-widget-content ui-corner-all ui-shadow': 0,
            'ui-dialog-rtl': 1, 'ui-dialog-draggable': 2 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dialog_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(),
            i0.ɵeld(0, [[3, 0], ['content', 1]], null, 4, 'div', [['class', 'ui-dialog-content ui-widget-content']], null, null, null, null, null)),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dialog_5)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Dialog_6)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_5 = _co.styleClass;
        var currVal_6 = _ck(_v, 6, 0, true, _co.rtl, _co.draggable);
        _ck(_v, 5, 0, currVal_5, currVal_6);
        var currVal_7 = _co.style;
        _ck(_v, 7, 0, currVal_7);
        var currVal_8 = _co.showHeader;
        _ck(_v, 10, 0, currVal_8);
        var currVal_9 = _co.contentStyle;
        _ck(_v, 13, 0, currVal_9);
        var currVal_10 = _co.footerFacet;
        _ck(_v, 19, 0, currVal_10);
        var currVal_11 = _co.resizable;
        _ck(_v, 22, 0, currVal_11);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.visible ? 'block' : 'none');
        var currVal_1 = _co.width;
        var currVal_2 = _co.height;
        var currVal_3 = _co.minWidth;
        var currVal_4 = (_co.visible ? 'visible' : 'hidden');
        _ck(_v, 4, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4);
    });
}
exports.View_Dialog_0 = View_Dialog_0;
function View_Dialog_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'p-dialog', [], null, null, null, View_Dialog_0, exports.RenderType_Dialog)),
        i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(12763136, null, 2, i1.Dialog, [i0.ElementRef, i4.DomHandler, i0.Renderer2], null, null), i0.ɵqud(335544320, 1, { headerFacet: 0 }), i0.ɵqud(335544320, 2, { footerFacet: 0 })], null, null);
}
exports.View_Dialog_Host_0 = View_Dialog_Host_0;
exports.DialogNgFactory = i0.ɵccf('p-dialog', i1.Dialog, View_Dialog_Host_0, { header: 'header', draggable: 'draggable', resizable: 'resizable',
    minWidth: 'minWidth', minHeight: 'minHeight', width: 'width', height: 'height', positionLeft: 'positionLeft',
    positionTop: 'positionTop', contentStyle: 'contentStyle', modal: 'modal', closeOnEscape: 'closeOnEscape',
    dismissableMask: 'dismissableMask', rtl: 'rtl', closable: 'closable', responsive: 'responsive',
    appendTo: 'appendTo', style: 'style', styleClass: 'styleClass', showHeader: 'showHeader',
    breakpoint: 'breakpoint', blockScroll: 'blockScroll', visible: 'visible' }, { onShow: 'onShow',
    onHide: 'onHide', visibleChange: 'visibleChange' }, ['p-header', '*', 'p-footer']);
