// primeng/components/organizationchart/organizationchart.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/organizationchart/organizationchart");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
var i4 = require("primeng/components/dom/domhandler");
exports.OrganizationChartModuleNgFactory = i0.ɵcmf(i1.OrganizationChartModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i1.OrganizationChartModule, i1.OrganizationChartModule, [])]);
});
var styles_OrganizationChart = [];
exports.RenderType_OrganizationChart = i0.ɵcrt({ encapsulation: 2,
    styles: styles_OrganizationChart, data: {} });
function View_OrganizationChart_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'table', [['class',
                'ui-organizationchart-table'], ['pOrganizationChartNode', '']], null, null, null, View_OrganizationChartNode_0, exports.RenderType_OrganizationChartNode)),
        i0.ɵdid(49152, null, 0, i1.OrganizationChartNode, [i1.OrganizationChart], { node: [0, 'node'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.root;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
function View_OrganizationChart_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrganizationChart_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-organizationchart ui-widget';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = _co.root;
        _ck(_v, 6, 0, currVal_3);
    }, null);
}
exports.View_OrganizationChart_0 = View_OrganizationChart_0;
function View_OrganizationChart_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'p-organizationChart', [], null, null, null, View_OrganizationChart_0, exports.RenderType_OrganizationChart)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(1097728, null, 1, i1.OrganizationChart, [i0.ElementRef,
            i4.DomHandler], null, null), i0.ɵqud(603979776, 1, { templates: 1 })], null, null);
}
exports.View_OrganizationChart_Host_0 = View_OrganizationChart_Host_0;
exports.OrganizationChartNgFactory = i0.ɵccf('p-organizationChart', i1.OrganizationChart, View_OrganizationChart_Host_0, { value: 'value', style: 'style',
    styleClass: 'styleClass', selectionMode: 'selectionMode', selection: 'selection' }, { selectionChange: 'selectionChange', onNodeSelect: 'onNodeSelect', onNodeUnselect: 'onNodeUnselect' }, []);
var styles_OrganizationChartNode = [];
exports.RenderType_OrganizationChartNode = i0.ɵcrt({ encapsulation: 2,
    styles: styles_OrganizationChartNode, data: { 'animation': [{ type: 7, name: 'childState',
                definitions: [{ type: 0, name: 'in', styles: { type: 6, styles: { opacity: 1 }, offset: null },
                        options: undefined }, { type: 1, expr: 'void => *', animation: [{ type: 6,
                                styles: { opacity: 0 }, offset: null }, { type: 4, styles: null,
                                timings: 150 }], options: null }, { type: 1, expr: '* => void', animation: [{ type: 4,
                                styles: { type: 6, styles: { opacity: 0 }, offset: null }, timings: 150 }],
                        options: null }], options: {} }] } });
function View_OrganizationChartNode_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.node.label;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_OrganizationChartNode_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'div', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(16777216, null, null, 1, 'p-organizationChartNodeTemplateLoader', [], null, null, null, View_OrganizationChartNodeTemplateLoader_0, exports.RenderType_OrganizationChartNodeTemplateLoader)), i0.ɵdid(245760, null, 0, i1.OrganizationChartNodeTemplateLoader, [i0.ViewContainerRef], { node: [0, 'node'],
            template: [1, 'template'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.node;
        var currVal_1 = _co.chart.getTemplateForNode(_co.node);
        _ck(_v, 3, 0, currVal_0, currVal_1);
    }, null);
}
function View_OrganizationChartNode_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'a', [['class',
                'ui-node-toggler'], ['href', '#']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggleNode($event, _co.node) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 2, 'i', [['class', 'fa ui-node-toggler-icon']], null, null, null, null, null)),
        i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        i0.ɵpod({ 'fa-chevron-down': 0, 'fa-chevron-up': 1 }), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'fa ui-node-toggler-icon';
        var currVal_1 = _ck(_v, 4, 0, _co.node.expanded, !_co.node.expanded);
        _ck(_v, 3, 0, currVal_0, currVal_1);
    }, null);
}
function View_OrganizationChartNode_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 18, 'tr', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 15, 'td', [], [[1, 'colspan', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵeld(0, null, null, 12, 'div', [], null, [[null,
                    'click']], function (_v, en, $event) {
                var ad = true;
                var _co = _v.component;
                if (('click' === en)) {
                    var pd_0 = (_co.onNodeClick($event, _co.node) !== false);
                    ad = (pd_0 && ad);
                }
                return ad;
            }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-organizationchart-selectable-node': 0,
            'ui-state-highlight': 1 }), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrganizationChartNode_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrganizationChartNode_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrganizationChartNode_4)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = i0.ɵinlineInterpolate(1, 'ui-organizationchart-node-content ui-widget-content ui-corner-all ', _co.node.styleClass, '');
        var currVal_2 = _ck(_v, 6, 0, (_co.chart.selectionMode && (_co.node.selectable !== false)), _co.isSelected());
        _ck(_v, 5, 0, currVal_1, currVal_2);
        var currVal_3 = !_co.chart.getTemplateForNode(_co.node);
        _ck(_v, 9, 0, currVal_3);
        var currVal_4 = _co.chart.getTemplateForNode(_co.node);
        _ck(_v, 12, 0, currVal_4);
        var currVal_5 = !_co.leaf;
        _ck(_v, 15, 0, currVal_5);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.colspan;
        _ck(_v, 2, 0, currVal_0);
    });
}
function View_OrganizationChartNode_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵeld(0, null, null, 3, 'td', [['class', 'ui-organizationchart-line-left']], null, null, null, null, null)),
        i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-organizationchart-line-top': 0 }), (_l()(), i0.ɵted(null, [' '])),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 3, 'td', [['class', 'ui-organizationchart-line-right']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-organizationchart-line-top': 0 }), (_l()(), i0.ɵted(null, [' '])),
        (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var currVal_0 = 'ui-organizationchart-line-left';
        var currVal_1 = _ck(_v, 3, 0, !_v.context.first);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = 'ui-organizationchart-line-right';
        var currVal_3 = _ck(_v, 8, 0, !_v.context.last);
        _ck(_v, 7, 0, currVal_2, currVal_3);
    }, null);
}
function View_OrganizationChartNode_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'td', [['colspan',
                '2']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 1, 'table', [['class', 'ui-organizationchart-table'], ['pOrganizationChartNode',
                '']], null, null, null, View_OrganizationChartNode_0, exports.RenderType_OrganizationChartNode)), i0.ɵdid(49152, null, 0, i1.OrganizationChartNode, [i1.OrganizationChart], { node: [0, 'node'] }, null), (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_OrganizationChartNode_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrganizationChartNode_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'tr', [['class', 'ui-organizationchart-lines']], [[4, 'visibility', null], [24, '@childState', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 3, 'td', [], [[1, 'colspan',
                0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 0, 'div', [['class', 'ui-organizationchart-line-down']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 4, 'tr', [['class', 'ui-organizationchart-lines']], [[4, 'visibility', null],
            [24, '@childState', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrganizationChartNode_5)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵeld(0, null, null, 4, 'tr', [['class', 'ui-organizationchart-nodes']], [[4, 'visibility', null], [24, '@childState', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrganizationChartNode_6)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0,
                'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.node;
        _ck(_v, 2, 0, currVal_0);
        var currVal_6 = _co.node.children;
        _ck(_v, 15, 0, currVal_6);
        var currVal_9 = _co.node.children;
        _ck(_v, 21, 0, currVal_9);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = ((!_co.leaf && _co.node.expanded) ? 'inherit' : 'hidden');
        var currVal_2 = 'in';
        _ck(_v, 4, 0, currVal_1, currVal_2);
        var currVal_3 = _co.colspan;
        _ck(_v, 6, 0, currVal_3);
        var currVal_4 = ((!_co.leaf && _co.node.expanded) ? 'inherit' : 'hidden');
        var currVal_5 = 'in';
        _ck(_v, 12, 0, currVal_4, currVal_5);
        var currVal_7 = ((!_co.leaf && _co.node.expanded) ? 'inherit' : 'hidden');
        var currVal_8 = 'in';
        _ck(_v, 18, 0, currVal_7, currVal_8);
    });
}
exports.View_OrganizationChartNode_0 = View_OrganizationChartNode_0;
function View_OrganizationChartNode_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['pOrganizationChartNode',
                '']], null, null, null, View_OrganizationChartNode_0, exports.RenderType_OrganizationChartNode)), i0.ɵdid(49152, null, 0, i1.OrganizationChartNode, [i1.OrganizationChart], null, null)], null, null);
}
exports.View_OrganizationChartNode_Host_0 = View_OrganizationChartNode_Host_0;
exports.OrganizationChartNodeNgFactory = i0.ɵccf('[pOrganizationChartNode]', i1.OrganizationChartNode, View_OrganizationChartNode_Host_0, { node: 'node', root: 'root',
    first: 'first', last: 'last' }, {}, []);
var styles_OrganizationChartNodeTemplateLoader = [];
exports.RenderType_OrganizationChartNodeTemplateLoader = i0.ɵcrt({ encapsulation: 2,
    styles: styles_OrganizationChartNodeTemplateLoader, data: {} });
function View_OrganizationChartNodeTemplateLoader_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_OrganizationChartNodeTemplateLoader_0 = View_OrganizationChartNodeTemplateLoader_0;
function View_OrganizationChartNodeTemplateLoader_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-organizationChartNodeTemplateLoader', [], null, null, null, View_OrganizationChartNodeTemplateLoader_0, exports.RenderType_OrganizationChartNodeTemplateLoader)), i0.ɵdid(245760, null, 0, i1.OrganizationChartNodeTemplateLoader, [i0.ViewContainerRef], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_OrganizationChartNodeTemplateLoader_Host_0 = View_OrganizationChartNodeTemplateLoader_Host_0;
exports.OrganizationChartNodeTemplateLoaderNgFactory = i0.ɵccf('p-organizationChartNodeTemplateLoader', i1.OrganizationChartNodeTemplateLoader, View_OrganizationChartNodeTemplateLoader_Host_0, { node: 'node', template: 'template' }, {}, []);
