// primeng/components/terminal/terminal.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/terminal/terminal");
var i2 = require("@angular/common");
var i3 = require("@angular/forms");
var i4 = require("primeng/components/dom/domhandler");
var i5 = require("primeng/components/terminal/terminalservice");
exports.TerminalModuleNgFactory = i0.ɵcmf(i1.TerminalModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(4608, i3.ɵi, i3.ɵi, []), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []),
        i0.ɵmpd(512, i3.ɵba, i3.ɵba, []), i0.ɵmpd(512, i3.FormsModule, i3.FormsModule, []), i0.ɵmpd(512, i1.TerminalModule, i1.TerminalModule, [])]);
});
var styles_Terminal = [];
exports.RenderType_Terminal = i0.ɵcrt({ encapsulation: 2, styles: styles_Terminal,
    data: {} });
function View_Terminal_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.welcomeMessage;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_Terminal_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 10, 'div', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(),
            i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-terminal-command']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['',
            ''])), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵeld(0, null, null, 1, 'div', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['',
            ''])), (_l()(), i0.ɵted(null, ['\n                ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.prompt;
        _ck(_v, 3, 0, currVal_0);
        var currVal_1 = _v.context.$implicit.text;
        _ck(_v, 6, 0, currVal_1);
        var currVal_2 = _v.context.$implicit.response;
        _ck(_v, 9, 0, currVal_2);
    });
}
function View_Terminal_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 25, 'div', [], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.focus(i0.ɵnov(_v, 19)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Terminal_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 4, 'div', [['class',
                'ui-terminal-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Terminal_2)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 11, 'div', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-terminal-content-prompt']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵeld(0, [['in', 1]], null, 5, 'input', [['autocomplete', 'off'], ['autofocus',
                    ''], ['class', 'ui-terminal-input'], ['type', 'text']], [[2, 'ng-untouched',
                    null], [2, 'ng-touched', null], [2, 'ng-pristine', null],
                [2, 'ng-dirty', null], [2, 'ng-valid', null], [2, 'ng-invalid',
                    null], [2, 'ng-pending', null]], [[null, 'ngModelChange'],
                [null, 'keydown'], [null, 'input'], [null, 'blur'],
                [null, 'compositionstart'], [null, 'compositionend']], function (_v, en, $event) {
                var ad = true;
                var _co = _v.component;
                if (('input' === en)) {
                    var pd_0 = (i0.ɵnov(_v, 20)._handleInput($event.target.value) !== false);
                    ad = (pd_0 && ad);
                }
                if (('blur' === en)) {
                    var pd_1 = (i0.ɵnov(_v, 20).onTouched() !== false);
                    ad = (pd_1 && ad);
                }
                if (('compositionstart' === en)) {
                    var pd_2 = (i0.ɵnov(_v, 20)._compositionStart() !== false);
                    ad = (pd_2 && ad);
                }
                if (('compositionend' === en)) {
                    var pd_3 = (i0.ɵnov(_v, 20)._compositionEnd($event.target.value) !== false);
                    ad = (pd_3 && ad);
                }
                if (('ngModelChange' === en)) {
                    var pd_4 = ((_co.command = $event) !== false);
                    ad = (pd_4 && ad);
                }
                if (('keydown' === en)) {
                    var pd_5 = (_co.handleCommand($event) !== false);
                    ad = (pd_5 && ad);
                }
                return ad;
            }, null, null)), i0.ɵdid(16384, null, 0, i3.DefaultValueAccessor, [i0.Renderer2, i0.ElementRef, [2, i3.COMPOSITION_BUFFER_MODE]], null, null), i0.ɵprd(1024, null, i3.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i3.DefaultValueAccessor]), i0.ɵdid(671744, null, 0, i3.NgModel, [[8, null],
            [8, null], [8, null], [2, i3.NG_VALUE_ACCESSOR]], { model: [0,
                'model'] }, { update: 'ngModelChange' }), i0.ɵprd(2048, null, i3.NgControl, null, [i3.NgModel]), i0.ɵdid(16384, null, 0, i3.NgControlStatus, [i3.NgControl], null, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-terminal ui-widget ui-widget-content ui-corner-all';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = _co.welcomeMessage;
        _ck(_v, 6, 0, currVal_3);
        var currVal_4 = _co.commands;
        _ck(_v, 11, 0, currVal_4);
        var currVal_13 = _co.command;
        _ck(_v, 22, 0, currVal_13);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_5 = _co.prompt;
        _ck(_v, 17, 0, currVal_5);
        var currVal_6 = i0.ɵnov(_v, 24).ngClassUntouched;
        var currVal_7 = i0.ɵnov(_v, 24).ngClassTouched;
        var currVal_8 = i0.ɵnov(_v, 24).ngClassPristine;
        var currVal_9 = i0.ɵnov(_v, 24).ngClassDirty;
        var currVal_10 = i0.ɵnov(_v, 24).ngClassValid;
        var currVal_11 = i0.ɵnov(_v, 24).ngClassInvalid;
        var currVal_12 = i0.ɵnov(_v, 24).ngClassPending;
        _ck(_v, 19, 0, currVal_6, currVal_7, currVal_8, currVal_9, currVal_10, currVal_11, currVal_12);
    });
}
exports.View_Terminal_0 = View_Terminal_0;
function View_Terminal_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-terminal', [], null, null, null, View_Terminal_0, exports.RenderType_Terminal)),
        i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(12763136, null, 0, i1.Terminal, [i0.ElementRef, i4.DomHandler, i5.TerminalService], null, null)], null, null);
}
exports.View_Terminal_Host_0 = View_Terminal_Host_0;
exports.TerminalNgFactory = i0.ɵccf('p-terminal', i1.Terminal, View_Terminal_Host_0, { welcomeMessage: 'welcomeMessage', prompt: 'prompt',
    style: 'style', styleClass: 'styleClass', response: 'response' }, {}, []);
