// primeng/components/inplace/inplace.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/inplace/inplace");
var i2 = require("@angular/common");
var i3 = require("primeng/components/button/button");
var i4 = require("primeng/components/dom/domhandler");
exports.InplaceModuleNgFactory = i0.ɵcmf(i1.InplaceModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.ButtonModule, i3.ButtonModule, []), i0.ɵmpd(512, i1.InplaceModule, i1.InplaceModule, [])]);
});
var styles_Inplace = [];
exports.RenderType_Inplace = i0.ɵcrt({ encapsulation: 2, styles: styles_Inplace,
    data: {} });
function View_Inplace_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'div', [['class',
                'ui-inplace-display']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.activate($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 }), (_l()(), i0.ɵted(null, ['\n                '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-inplace-display';
        var currVal_1 = _ck(_v, 2, 0, _co.disabled);
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Inplace_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'button', [['icon',
                'fa-close'], ['pButton', ''], ['type', 'button']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.deactivate($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(4341760, null, 0, i3.Button, [i0.ElementRef, i4.DomHandler], { icon: [0, 'icon'] }, null)], function (_ck, _v) {
        var currVal_0 = 'fa-close';
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
function View_Inplace_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 6, 'div', [['class',
                'ui-inplace-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Inplace_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.closable;
        _ck(_v, 5, 0, currVal_0);
    }, null);
}
function View_Inplace_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 9, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Inplace_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Inplace_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-inplace ui-widget';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = !_co.active;
        _ck(_v, 6, 0, currVal_3);
        var currVal_4 = _co.active;
        _ck(_v, 9, 0, currVal_4);
    }, null);
}
exports.View_Inplace_0 = View_Inplace_0;
function View_Inplace_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-inplace', [], null, null, null, View_Inplace_0, exports.RenderType_Inplace)),
        i0.ɵdid(49152, null, 0, i1.Inplace, [], null, null)], null, null);
}
exports.View_Inplace_Host_0 = View_Inplace_Host_0;
exports.InplaceNgFactory = i0.ɵccf('p-inplace', i1.Inplace, View_Inplace_Host_0, { active: 'active', closable: 'closable', disabled: 'disabled',
    style: 'style', styleClass: 'styleClass' }, { onActivate: 'onActivate', onDeactivate: 'onDeactivate' }, ['[pInplaceDisplay]', '[pInplaceContent]']);
var styles_InplaceDisplay = [];
exports.RenderType_InplaceDisplay = i0.ɵcrt({ encapsulation: 2,
    styles: styles_InplaceDisplay, data: {} });
function View_InplaceDisplay_0(_l) {
    return i0.ɵvid(0, [i0.ɵncd(null, 0)], null, null);
}
exports.View_InplaceDisplay_0 = View_InplaceDisplay_0;
function View_InplaceDisplay_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-inplaceDisplay', [], null, null, null, View_InplaceDisplay_0, exports.RenderType_InplaceDisplay)), i0.ɵdid(49152, null, 0, i1.InplaceDisplay, [], null, null)], null, null);
}
exports.View_InplaceDisplay_Host_0 = View_InplaceDisplay_Host_0;
exports.InplaceDisplayNgFactory = i0.ɵccf('p-inplaceDisplay', i1.InplaceDisplay, View_InplaceDisplay_Host_0, {}, {}, ['*']);
var styles_InplaceContent = [];
exports.RenderType_InplaceContent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_InplaceContent, data: {} });
function View_InplaceContent_0(_l) {
    return i0.ɵvid(0, [i0.ɵncd(null, 0)], null, null);
}
exports.View_InplaceContent_0 = View_InplaceContent_0;
function View_InplaceContent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-inplaceContent', [], null, null, null, View_InplaceContent_0, exports.RenderType_InplaceContent)), i0.ɵdid(49152, null, 0, i1.InplaceContent, [], null, null)], null, null);
}
exports.View_InplaceContent_Host_0 = View_InplaceContent_Host_0;
exports.InplaceContentNgFactory = i0.ɵccf('p-inplaceContent', i1.InplaceContent, View_InplaceContent_Host_0, {}, {}, ['*']);
