// primeng/components/selectbutton/selectbutton.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/selectbutton/selectbutton");
var i2 = require("@angular/common");
var i3 = require("@angular/forms");
exports.SelectButtonModuleNgFactory = i0.ɵcmf(i1.SelectButtonModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.SelectButtonModule, i1.SelectButtonModule, [])]);
});
var styles_SelectButton = [];
exports.RenderType_SelectButton = i0.ɵcrt({ encapsulation: 2, styles: styles_SelectButton,
    data: {} });
function View_SelectButton_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 11, 'div', [], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onItemClick($event, _v.context.$implicit, i0.ɵnov(_v, 9), _v.context.index) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-active': 0, 'ui-state-disabled': 1,
            'ui-state-focus': 2 }), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-button-text ui-clickable']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'ui-helper-hidden-accessible']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵeld(0, [['cbox', 1]], null, 0, 'input', [['type', 'checkbox']], [[8, 'checked',
                0], [1, 'tabindex', 0], [1, 'disabled', 0]], [[null, 'focus'], [null,
                'blur']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('focus' === en)) {
                var pd_0 = (_co.onFocus($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('blur' === en)) {
                var pd_1 = (_co.onBlur($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = i0.ɵinlineInterpolate(1, 'ui-button ui-widget ui-state-default ui-button-text-only ', _v.context.$implicit.styleClass, '');
        var currVal_1 = _ck(_v, 2, 0, _co.isSelected(_v.context.$implicit), _co.disabled, (i0.ɵnov(_v, 9) == _co.focusedItem));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = _v.context.$implicit.label;
        _ck(_v, 5, 0, currVal_2);
        var currVal_3 = _co.isSelected(_v.context.$implicit);
        var currVal_4 = _co.tabindex;
        var currVal_5 = _co.disabled;
        _ck(_v, 9, 0, currVal_3, currVal_4, currVal_5);
    });
}
function View_SelectButton_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_SelectButton_1)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = ('ui-selectbutton ui-buttonset ui-widget ui-corner-all ui-buttonset-' + _co.options.length);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = _co.options;
        _ck(_v, 6, 0, currVal_3);
    }, null);
}
exports.View_SelectButton_0 = View_SelectButton_0;
function View_SelectButton_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-selectButton', [], null, null, null, View_SelectButton_0, exports.RenderType_SelectButton)), i0.ɵprd(5120, null, i3.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i1.SelectButton]), i0.ɵdid(49152, null, 0, i1.SelectButton, [], null, null)], null, null);
}
exports.View_SelectButton_Host_0 = View_SelectButton_Host_0;
exports.SelectButtonNgFactory = i0.ɵccf('p-selectButton', i1.SelectButton, View_SelectButton_Host_0, { options: 'options', tabindex: 'tabindex',
    multiple: 'multiple', style: 'style', styleClass: 'styleClass', disabled: 'disabled' }, { onOptionClick: 'onOptionClick', onChange: 'onChange' }, []);
