// primeng/components/toolbar/toolbar.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/toolbar/toolbar");
var i2 = require("@angular/common");
exports.ToolbarModuleNgFactory = i0.ɵcmf(i1.ToolbarModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.ToolbarModule, i1.ToolbarModule, [])]);
});
var styles_Toolbar = [];
exports.RenderType_Toolbar = i0.ɵcrt({ encapsulation: 2, styles: styles_Toolbar,
    data: {} });
function View_Toolbar_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 5, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-toolbar ui-widget ui-widget-header ui-corner-all ui-helper-clearfix';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
    }, null);
}
exports.View_Toolbar_0 = View_Toolbar_0;
function View_Toolbar_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-toolbar', [], null, null, null, View_Toolbar_0, exports.RenderType_Toolbar)),
        i0.ɵdid(49152, null, 0, i1.Toolbar, [], null, null)], null, null);
}
exports.View_Toolbar_Host_0 = View_Toolbar_Host_0;
exports.ToolbarNgFactory = i0.ɵccf('p-toolbar', i1.Toolbar, View_Toolbar_Host_0, { style: 'style', styleClass: 'styleClass' }, {}, ['*']);
