// primeng/components/treetable/treetable.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/treetable/treetable");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
var i4 = require("../common/shared.ngfactory");
var i5 = require("primeng/components/dom/domhandler");
exports.TreeTableModuleNgFactory = i0.ɵcmf(i1.TreeTableModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i1.TreeTableModule, i1.TreeTableModule, [])]);
});
var styles_TreeTable = [];
exports.RenderType_TreeTable = i0.ɵcrt({ encapsulation: 2, styles: styles_TreeTable,
    data: {} });
function View_TreeTable_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-treetable-header ui-widget-header']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_TreeTable_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-column-title']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.header;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_TreeTable_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'span', [['class',
                'ui-column-title']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                                    '])),
        (_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnHeaderTemplateLoader', [], null, null, null, i4.View_ColumnHeaderTemplateLoader_0, i4.RenderType_ColumnHeaderTemplateLoader)), i0.ɵdid(245760, null, 0, i3.ColumnHeaderTemplateLoader, [i0.ViewContainerRef], { column: [0, 'column'] }, null), (_l()(), i0.ɵted(null, ['\n                                ']))], function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_TreeTable_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, [['headerCell', 1]], null, 9, 'th', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0,
                'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_4)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                            ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.styleClass;
        var currVal_1 = 'ui-state-default ui-unselectable-text';
        _ck(_v, 1, 0, currVal_0, currVal_1);
        var currVal_2 = _v.context.$implicit.style;
        _ck(_v, 2, 0, currVal_2);
        var currVal_3 = !_v.context.$implicit.headerTemplate;
        _ck(_v, 5, 0, currVal_3);
        var currVal_4 = _v.context.$implicit.headerTemplate;
        _ck(_v, 8, 0, currVal_4);
    }, null);
}
function View_TreeTable_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-column-footer']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.footer;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_TreeTable_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'span', [['class',
                'ui-column-footer']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                                    '])),
        (_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnFooterTemplateLoader', [], null, null, null, i4.View_ColumnFooterTemplateLoader_0, i4.RenderType_ColumnFooterTemplateLoader)), i0.ɵdid(245760, null, 0, i3.ColumnFooterTemplateLoader, [i0.ViewContainerRef], { column: [0, 'column'] }, null), (_l()(), i0.ɵted(null, ['\n                                ']))], function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_TreeTable_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 10, 'td', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-state-default': 0 }),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_7)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n                                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_TreeTable_8)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                            ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.styleClass;
        var currVal_1 = _ck(_v, 2, 0, true);
        _ck(_v, 1, 0, currVal_0, currVal_1);
        var currVal_2 = _v.context.$implicit.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = !_v.context.$implicit.footerTemplate;
        _ck(_v, 6, 0, currVal_3);
        var currVal_4 = _v.context.$implicit.footerTemplate;
        _ck(_v, 9, 0, currVal_4);
    }, null);
}
function View_TreeTable_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 7, 'tfoot', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(0, null, null, 4, 'tr', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_6)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.columns;
        _ck(_v, 5, 0, currVal_0);
    }, null);
}
function View_TreeTable_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'tbody', [['class',
                'ui-treetable-data ui-widget-content'], ['pTreeRow', '']], null, null, null, View_UITreeRow_0, exports.RenderType_UITreeRow)), i0.ɵdid(114688, null, 0, i1.UITreeRow, [i1.TreeTable], { node: [0, 'node'], level: [1, 'level'], labelExpand: [2,
                'labelExpand'], labelCollapse: [3, 'labelCollapse'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit;
        var currVal_1 = 0;
        var currVal_2 = _co.labelExpand;
        var currVal_3 = _co.labelCollapse;
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2, currVal_3);
    }, null);
}
function View_TreeTable_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-treetable-footer ui-widget-header']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_TreeTable_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { tableViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 31, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0,
                'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 20, 'div', [['class', 'ui-treetable-tablewrapper']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, [[1,
                0], ['tbl', 1]], null, 17, 'table', [['class', 'ui-widget-content']], [[8,
                'className', 0]], null, null, null, null)),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵeld(0, null, null, 7, 'thead', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 4, 'tr', [['class', 'ui-state-default']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_2)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_5)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_9)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            \n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TreeTable_10)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-treetable ui-widget';
        _ck(_v, 3, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 4, 0, currVal_2);
        var currVal_3 = _co.header;
        _ck(_v, 7, 0, currVal_3);
        var currVal_5 = _co.tableStyle;
        _ck(_v, 12, 0, currVal_5);
        var currVal_6 = _co.columns;
        _ck(_v, 19, 0, currVal_6);
        var currVal_7 = _co.hasFooter();
        _ck(_v, 24, 0, currVal_7);
        var currVal_8 = _co.value;
        _ck(_v, 27, 0, currVal_8);
        var currVal_9 = _co.footer;
        _ck(_v, 32, 0, currVal_9);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_4 = _co.tableStyleClass;
        _ck(_v, 11, 0, currVal_4);
    });
}
exports.View_TreeTable_0 = View_TreeTable_0;
function View_TreeTable_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'p-treeTable', [], null, null, null, View_TreeTable_0, exports.RenderType_TreeTable)),
        i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []), i0.ɵdid(1097728, null, 3, i1.TreeTable, [i0.ElementRef, i5.DomHandler, i0.ChangeDetectorRef,
            i0.Renderer2], null, null), i0.ɵqud(335544320, 1, { header: 0 }),
        i0.ɵqud(335544320, 2, { footer: 0 }), i0.ɵqud(603979776, 3, { cols: 1 })], null, null);
}
exports.View_TreeTable_Host_0 = View_TreeTable_Host_0;
exports.TreeTableNgFactory = i0.ɵccf('p-treeTable', i1.TreeTable, View_TreeTable_Host_0, { value: 'value', selectionMode: 'selectionMode',
    selection: 'selection', style: 'style', styleClass: 'styleClass', labelExpand: 'labelExpand',
    labelCollapse: 'labelCollapse', metaKeySelection: 'metaKeySelection', contextMenu: 'contextMenu',
    toggleColumnIndex: 'toggleColumnIndex', tableStyle: 'tableStyle', tableStyleClass: 'tableStyleClass',
    collapsedIcon: 'collapsedIcon', expandedIcon: 'expandedIcon' }, { onRowDblclick: 'onRowDblclick',
    selectionChange: 'selectionChange', onNodeSelect: 'onNodeSelect', onNodeUnselect: 'onNodeUnselect',
    onNodeExpand: 'onNodeExpand', onNodeCollapse: 'onNodeCollapse', onContextMenuSelect: 'onContextMenuSelect' }, ['p-header', 'p-footer']);
var styles_UITreeRow = [];
exports.RenderType_UITreeRow = i0.ɵcrt({ encapsulation: 2, styles: styles_UITreeRow,
    data: {} });
function View_UITreeRow_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'a', [['class',
                'ui-treetable-toggler fa fa-fw ui-clickable'], ['href', '#']], [[8, 'title', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggle($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), i0.ɵpod({ 'margin-left': 0,
            'visibility': 1 }), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = 'ui-treetable-toggler fa fa-fw ui-clickable';
        var currVal_2 = (_co.node.expanded ? _co.treeTable.expandedIcon : _co.treeTable.collapsedIcon);
        _ck(_v, 1, 0, currVal_1, currVal_2);
        var currVal_3 = _ck(_v, 3, 0, ((_co.level * 16) + 'px'), (_co.isLeaf() ? 'hidden' : 'visible'));
        _ck(_v, 2, 0, currVal_3);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.node.expanded ? _co.labelCollapse : _co.labelExpand);
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_UITreeRow_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'div', [['class',
                'ui-chkbox ui-treetable-checkbox']], null, null, null, null, null)), (_l()(), i0.ɵeld(0, null, null, 4, 'div', [['class', 'ui-chkbox-box ui-widget ui-corner-all ui-state-default']], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class', 'ui-chkbox-icon ui-clickable fa']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'fa-check': 0,
            'fa-minus': 1 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-chkbox-icon ui-clickable fa';
        var currVal_1 = _ck(_v, 5, 0, _co.isSelected(), _co.node.partialSelected);
        _ck(_v, 4, 0, currVal_0, currVal_1);
    }, null);
}
function View_UITreeRow_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.resolveFieldData(_co.node.data, _v.parent.context.$implicit.field);
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_UITreeRow_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnBodyTemplateLoader', [], null, null, null, i4.View_ColumnBodyTemplateLoader_0, i4.RenderType_ColumnBodyTemplateLoader)), i0.ɵdid(770048, null, 0, i3.ColumnBodyTemplateLoader, [i0.ViewContainerRef], { column: [0, 'column'], rowData: [1, 'rowData'] }, null),
        (_l()(), i0.ɵand(0, null, null, 0))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.parent.context.$implicit;
        var currVal_1 = _co.node;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_UITreeRow_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 13, 'td', [], [[8, 'className', 0]], [[null, 'click'], [null, 'dblclick'], [null,
                'touchend'], [null, 'contextmenu']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onRowClick($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('dblclick' === en)) {
                var pd_1 = (_co.rowDblClick($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('touchend' === en)) {
                var pd_2 = (_co.onRowTouchEnd() !== false);
                ad = (pd_2 && ad);
            }
            if (('contextmenu' === en)) {
                var pd_3 = (_co.onRowRightClick($event) !== false);
                ad = (pd_3 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_UITreeRow_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_UITreeRow_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_UITreeRow_4)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_UITreeRow_5)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _v.context.$implicit.style;
        _ck(_v, 1, 0, currVal_1);
        var currVal_2 = (_v.context.index == _co.treeTable.toggleColumnIndex);
        _ck(_v, 4, 0, currVal_2);
        var currVal_3 = ((_co.treeTable.selectionMode == 'checkbox') && (_v.context.index == 0));
        _ck(_v, 7, 0, currVal_3);
        var currVal_4 = !_v.context.$implicit.template;
        _ck(_v, 9, 0, currVal_4);
        var currVal_5 = _v.context.$implicit.template;
        _ck(_v, 12, 0, currVal_5);
    }, function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.styleClass;
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_UITreeRow_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'tbody', [['pTreeRow',
                '']], null, null, null, View_UITreeRow_0, exports.RenderType_UITreeRow)),
        i0.ɵdid(114688, null, 0, i1.UITreeRow, [i1.TreeTable], { node: [0, 'node'],
            parentNode: [1, 'parentNode'], level: [2, 'level'], labelExpand: [3, 'labelExpand'],
            labelCollapse: [4, 'labelCollapse'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit;
        var currVal_1 = _co.node;
        var currVal_2 = (_co.level + 1);
        var currVal_3 = _co.labelExpand;
        var currVal_4 = _co.labelCollapse;
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4);
    }, null);
}
function View_UITreeRow_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 11, 'div', [['class',
                'ui-treetable-row'], ['style', 'display:table-row']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 8, 'td', [['class', 'ui-treetable-child-table-container']], [[1, 'colspan', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 5, 'table', [], [[8, 'className', 0]], null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0,
                'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_UITreeRow_7)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n        ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = _co.treeTable.tableStyle;
        _ck(_v, 5, 0, currVal_2);
        var currVal_3 = _co.node.children;
        _ck(_v, 8, 0, currVal_3);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.treeTable.columns.length;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = _co.treeTable.tableStyleClass;
        _ck(_v, 4, 0, currVal_1);
    });
}
function View_UITreeRow_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'div', [['class', 'ui-treetable-row']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-state-highlight': 0,
            'ui-treetable-row-selectable': 1 }), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_UITreeRow_1)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_UITreeRow_6)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-treetable-row';
        var currVal_1 = _ck(_v, 3, 0, _co.isSelected(), (_co.treeTable.selectionMode && (_co.node.selectable !== false)));
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.treeTable.columns;
        _ck(_v, 6, 0, currVal_2);
        var currVal_3 = (_co.node.children && _co.node.expanded);
        _ck(_v, 10, 0, currVal_3);
    }, null);
}
exports.View_UITreeRow_0 = View_UITreeRow_0;
function View_UITreeRow_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['pTreeRow',
                '']], null, null, null, View_UITreeRow_0, exports.RenderType_UITreeRow)),
        i0.ɵdid(114688, null, 0, i1.UITreeRow, [i1.TreeTable], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_UITreeRow_Host_0 = View_UITreeRow_Host_0;
exports.UITreeRowNgFactory = i0.ɵccf('[pTreeRow]', i1.UITreeRow, View_UITreeRow_Host_0, { node: 'node', parentNode: 'parentNode', level: 'level',
    labelExpand: 'labelExpand', labelCollapse: 'labelCollapse' }, {}, []);
