// @ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation-select.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@angular/forms");
var i2 = require("@ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation-select");
var i3 = require("@angular/common");
var i4 = require("@ng-bootstrap/ng-bootstrap/datepicker/datepicker-i18n");
var i5 = require("@ng-bootstrap/ng-bootstrap/datepicker/ngb-calendar");
var styles_NgbDatepickerNavigationSelect = ['select[_ngcontent-%COMP%] {\n      \n      padding: 0.25rem 0.5rem;\n      font-size: 0.875rem;      \n      line-height: 1.25;\n      \n      height: inherit;\n      width: 50%;\n    }'];
exports.RenderType_NgbDatepickerNavigationSelect = i0.ɵcrt({ encapsulation: 0,
    styles: styles_NgbDatepickerNavigationSelect, data: {} });
function View_NgbDatepickerNavigationSelect_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'option', [], null, null, null, null, null)), i0.ɵdid(147456, null, 0, i1.NgSelectOption, [i0.ElementRef, i0.Renderer2, [8, null]], { value: [0, 'value'] }, null), i0.ɵdid(147456, null, 0, i1.ɵq, [i0.ElementRef,
            i0.Renderer2, [8, null]], { value: [0, 'value'] }, null), (_l()(), i0.ɵted(null, ['', '']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit;
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = _v.context.$implicit;
        _ck(_v, 2, 0, currVal_1);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = _co.i18n.getMonthShortName(_v.context.$implicit);
        _ck(_v, 3, 0, currVal_2);
    });
}
function View_NgbDatepickerNavigationSelect_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'option', [], null, null, null, null, null)), i0.ɵdid(147456, null, 0, i1.NgSelectOption, [i0.ElementRef, i0.Renderer2, [8, null]], { value: [0, 'value'] }, null), i0.ɵdid(147456, null, 0, i1.ɵq, [i0.ElementRef,
            i0.Renderer2, [8, null]], { value: [0, 'value'] }, null), (_l()(), i0.ɵted(null, ['', '']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit;
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = _v.context.$implicit;
        _ck(_v, 2, 0, currVal_1);
    }, function (_ck, _v) {
        var currVal_2 = _v.context.$implicit;
        _ck(_v, 3, 0, currVal_2);
    });
}
function View_NgbDatepickerNavigationSelect_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 4, 'select', [['class', 'custom-select d-inline-block'], ['tabindex',
                '-1']], [[8, 'disabled', 0], [8, 'value', 0]], [[null, 'change']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('change' === en)) {
                var pd_0 = (_co.changeMonth($event.target.value) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbDatepickerNavigationSelect_1)),
        i0.ɵdid(802816, null, 0, i3.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 4, 'select', [['class',
                'custom-select d-inline-block'], ['tabindex', '-1']], [[8, 'disabled', 0], [8,
                'value', 0]], [[null, 'change']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('change' === en)) {
                var pd_0 = (_co.changeYear($event.target.value) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbDatepickerNavigationSelect_2)),
        i0.ɵdid(802816, null, 0, i3.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵted(null, [' \n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = _co.months;
        _ck(_v, 4, 0, currVal_2);
        var currVal_5 = _co.years;
        _ck(_v, 9, 0, currVal_5);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.disabled;
        var currVal_1 = ((_co.date == null) ? null : _co.date.month);
        _ck(_v, 1, 0, currVal_0, currVal_1);
        var currVal_3 = _co.disabled;
        var currVal_4 = ((_co.date == null) ? null : _co.date.year);
        _ck(_v, 6, 0, currVal_3, currVal_4);
    });
}
exports.View_NgbDatepickerNavigationSelect_0 = View_NgbDatepickerNavigationSelect_0;
function View_NgbDatepickerNavigationSelect_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'ngb-datepicker-navigation-select', [], null, null, null, View_NgbDatepickerNavigationSelect_0, exports.RenderType_NgbDatepickerNavigationSelect)), i0.ɵdid(573440, null, 0, i2.NgbDatepickerNavigationSelect, [i4.NgbDatepickerI18n, i5.NgbCalendar], null, null)], null, null);
}
exports.View_NgbDatepickerNavigationSelect_Host_0 = View_NgbDatepickerNavigationSelect_Host_0;
exports.NgbDatepickerNavigationSelectNgFactory = i0.ɵccf('ngb-datepicker-navigation-select', i2.NgbDatepickerNavigationSelect, View_NgbDatepickerNavigationSelect_Host_0, { date: 'date',
    disabled: 'disabled', maxDate: 'maxDate', minDate: 'minDate' }, { select: 'select' }, []);
