// @ng-bootstrap/ng-bootstrap/datepicker/datepicker-day-view.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@ng-bootstrap/ng-bootstrap/datepicker/datepicker-day-view");
var styles_NgbDatepickerDayView = ['[_nghost-%COMP%] {\n      text-align: center;\n      width: 2rem;\n      height: 2rem;\n      line-height: 2rem;\n      border-radius: 0.25rem;\n      background: transparent;\n    }\n    .outside[_nghost-%COMP%] {\n      opacity: 0.5;\n    }'];
exports.RenderType_NgbDatepickerDayView = i0.ɵcrt({ encapsulation: 0,
    styles: styles_NgbDatepickerDayView, data: {} });
function View_NgbDatepickerDayView_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.date.day;
        _ck(_v, 0, 0, currVal_0);
    });
}
exports.View_NgbDatepickerDayView_0 = View_NgbDatepickerDayView_0;
function View_NgbDatepickerDayView_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['class',
                'btn-light'], ['ngbDatepickerDayView', '']], [[2, 'bg-primary', null], [2,
                'text-white', null], [2, 'text-muted', null], [2, 'outside', null],
            [2, 'active', null]], null, null, View_NgbDatepickerDayView_0, exports.RenderType_NgbDatepickerDayView)), i0.ɵdid(49152, null, 0, i1.NgbDatepickerDayView, [], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).selected;
        var currVal_1 = i0.ɵnov(_v, 1).selected;
        var currVal_2 = i0.ɵnov(_v, 1).isMuted();
        var currVal_3 = i0.ɵnov(_v, 1).isMuted();
        var currVal_4 = i0.ɵnov(_v, 1).focused;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4);
    });
}
exports.View_NgbDatepickerDayView_Host_0 = View_NgbDatepickerDayView_Host_0;
exports.NgbDatepickerDayViewNgFactory = i0.ɵccf('[ngbDatepickerDayView]', i1.NgbDatepickerDayView, View_NgbDatepickerDayView_Host_0, { currentMonth: 'currentMonth',
    date: 'date', disabled: 'disabled', focused: 'focused', selected: 'selected' }, {}, []);
