// primeng/components/datalist/datalist.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/datalist/datalist");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
var i4 = require("primeng/components/paginator/paginator");
var i5 = require("../paginator/paginator.ngfactory");
exports.DataListModuleNgFactory = i0.ɵcmf(i1.DataListModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i4.PaginatorModule, i4.PaginatorModule, []),
        i0.ɵmpd(512, i1.DataListModule, i1.DataListModule, [])]);
});
var styles_DataList = [];
exports.RenderType_DataList = i0.ɵcrt({ encapsulation: 2, styles: styles_DataList,
    data: {} });
function View_DataList_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-datalist-header ui-widget-header ui-corner-top']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_DataList_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-paginator', [['styleClass', 'ui-paginator-bottom']], null, [[null, 'onPageChange']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('onPageChange' === en)) {
                var pd_0 = (_co.paginate($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, i5.View_Paginator_0, i5.RenderType_Paginator)), i0.ɵdid(49152, null, 0, i4.Paginator, [], { pageLinkSize: [0, 'pageLinkSize'], styleClass: [1,
                'styleClass'], rowsPerPageOptions: [2, 'rowsPerPageOptions'], alwaysShow: [3,
                'alwaysShow'], totalRecords: [4, 'totalRecords'], first: [5, 'first'], rows: [6,
                'rows'] }, { onPageChange: 'onPageChange' })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.pageLinks;
        var currVal_1 = 'ui-paginator-bottom';
        var currVal_2 = _co.rowsPerPageOptions;
        var currVal_3 = _co.alwaysShowPaginator;
        var currVal_4 = _co.totalRecords;
        var currVal_5 = _co.first;
        var currVal_6 = _co.rows;
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6);
    }, null);
}
function View_DataList_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['class',
                'ui-datalist-emptymessage']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.emptyMessage;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_DataList_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_DataList_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'li', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DataList_5)), i0.ɵdid(212992, null, 0, i3.TemplateWrapper, [i0.ViewContainerRef], { item: [0, 'item'], index: [1,
                'index'], templateRef: [2, 'templateRef'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit;
        var currVal_1 = _v.context.index;
        var currVal_2 = _co.itemTemplate;
        _ck(_v, 3, 0, currVal_0, currVal_1, currVal_2);
    }, null);
}
function View_DataList_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-paginator', [['styleClass', 'ui-paginator-bottom']], null, [[null, 'onPageChange']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('onPageChange' === en)) {
                var pd_0 = (_co.paginate($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, i5.View_Paginator_0, i5.RenderType_Paginator)), i0.ɵdid(49152, null, 0, i4.Paginator, [], { pageLinkSize: [0, 'pageLinkSize'], styleClass: [1,
                'styleClass'], rowsPerPageOptions: [2, 'rowsPerPageOptions'], alwaysShow: [3,
                'alwaysShow'], totalRecords: [4, 'totalRecords'], first: [5, 'first'], rows: [6,
                'rows'] }, { onPageChange: 'onPageChange' })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.pageLinks;
        var currVal_1 = 'ui-paginator-bottom';
        var currVal_2 = _co.rowsPerPageOptions;
        var currVal_3 = _co.alwaysShowPaginator;
        var currVal_4 = _co.totalRecords;
        var currVal_5 = _co.first;
        var currVal_6 = _co.rows;
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6);
    }, null);
}
function View_DataList_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-datalist-footer ui-widget-header ui-corner-bottom']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_DataList_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 30, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-datalist ui-widget': 0,
            'ui-datalist-scrollable': 1 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DataList_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DataList_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 12, 'div', [['class', 'ui-datalist-content ui-widget-content']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null),
        i0.ɵpod({ 'max-height': 0 }), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DataList_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, null, null, 4, 'ul', [['class', 'ui-datalist-data']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DataList_4)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'], ngForTrackBy: [1, 'ngForTrackBy'] }, null), (_l()(),
            i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DataList_6)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DataList_7)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 3, 0, true, _co.scrollable);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 4, 0, currVal_2);
        var currVal_3 = _co.header;
        _ck(_v, 7, 0, currVal_3);
        var currVal_4 = ((_co.paginator && (_co.paginatorPosition != 'bottom')) || (_co.paginatorPosition == 'both'));
        _ck(_v, 10, 0, currVal_4);
        var currVal_5 = _ck(_v, 14, 0, _co.scrollHeight);
        _ck(_v, 13, 0, currVal_5);
        var currVal_6 = _co.isEmpty();
        _ck(_v, 17, 0, currVal_6);
        var currVal_7 = _co.dataToRender;
        var currVal_8 = _co.trackBy;
        _ck(_v, 22, 0, currVal_7, currVal_8);
        var currVal_9 = ((_co.paginator && (_co.paginatorPosition != 'top')) || (_co.paginatorPosition == 'both'));
        _ck(_v, 27, 0, currVal_9);
        var currVal_10 = _co.footer;
        _ck(_v, 30, 0, currVal_10);
    }, null);
}
exports.View_DataList_0 = View_DataList_0;
function View_DataList_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'p-dataList', [], null, null, null, View_DataList_0, exports.RenderType_DataList)),
        i0.ɵdid(5554176, null, 3, i1.DataList, [i0.ElementRef, i0.IterableDiffers], null, null), i0.ɵqud(335544320, 1, { header: 0 }), i0.ɵqud(335544320, 2, { footer: 0 }), i0.ɵqud(603979776, 3, { templates: 1 })], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_DataList_Host_0 = View_DataList_Host_0;
exports.DataListNgFactory = i0.ɵccf('p-dataList', i1.DataList, View_DataList_Host_0, { paginator: 'paginator', rows: 'rows', totalRecords: 'totalRecords',
    pageLinks: 'pageLinks', rowsPerPageOptions: 'rowsPerPageOptions', lazy: 'lazy',
    style: 'style', styleClass: 'styleClass', paginatorPosition: 'paginatorPosition',
    emptyMessage: 'emptyMessage', alwaysShowPaginator: 'alwaysShowPaginator', trackBy: 'trackBy',
    immutable: 'immutable', scrollable: 'scrollable', scrollHeight: 'scrollHeight',
    value: 'value' }, { onLazyLoad: 'onLazyLoad', onPage: 'onPage' }, ['p-header', 'p-footer']);
