// primeng/components/menubar/menubar.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/menubar/menubar");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
var i4 = require("primeng/components/dom/domhandler");
exports.MenubarModuleNgFactory = i0.ɵcmf(i1.MenubarModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.RouterModule, i3.RouterModule, [[2, i3.ɵa], [2, i3.Router]]), i0.ɵmpd(512, i1.MenubarModule, i1.MenubarModule, [])]);
});
var styles_Menubar = [];
exports.RenderType_Menubar = i0.ɵcrt({ encapsulation: 2, styles: styles_Menubar,
    data: {} });
function View_Menubar_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 11, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-menubar ui-menu ui-widget ui-widget-content ui-corner-all ui-helper-clearfix': 0 }),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 5, 'p-menubarSub', [['root', 'root']], null, null, null, View_MenubarSub_0, exports.RenderType_MenubarSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.MenubarSub, [i4.DomHandler], { item: [0, 'item'], root: [1, 'root'] }, null), (_l()(), i0.ɵted(0, ['\n                '])),
        i0.ɵncd(0, 0), (_l()(), i0.ɵted(0, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 3, 0, true);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 4, 0, currVal_2);
        var currVal_3 = _co.model;
        var currVal_4 = 'root';
        _ck(_v, 8, 0, currVal_3, currVal_4);
    }, null);
}
exports.View_Menubar_0 = View_Menubar_0;
function View_Menubar_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-menubar', [], null, null, null, View_Menubar_0, exports.RenderType_Menubar)),
        i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.Menubar, [i0.ElementRef, i4.DomHandler, i0.Renderer2], null, null)], null, null);
}
exports.View_Menubar_Host_0 = View_Menubar_Host_0;
exports.MenubarNgFactory = i0.ɵccf('p-menubar', i1.Menubar, View_Menubar_Host_0, { model: 'model', style: 'style', styleClass: 'styleClass' }, {}, ['*']);
var styles_MenubarSub = [];
exports.RenderType_MenubarSub = i0.ɵcrt({ encapsulation: 2, styles: styles_MenubarSub,
    data: {} });
function View_MenubarSub_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'li', [['class',
                'ui-menu-separator ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                ']))], null, null);
}
function View_MenubarSub_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _v.parent.parent.parent.context.$implicit.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_MenubarSub_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class',
                'ui-submenu-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'fa-caret-down': 0, 'fa-caret-right': 1 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-submenu-icon fa fa-fw';
        var currVal_1 = _ck(_v, 2, 0, _co.root, !_co.root);
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_MenubarSub_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 13, 'a', [], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.itemClick($event, _v.parent.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-menuitem-link ui-corner-all': 0, 'ui-state-disabled': 1 }),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_5)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_6)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_3 = _v.parent.parent.context.$implicit.styleClass;
        var currVal_4 = _ck(_v, 2, 0, true, _v.parent.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_3, currVal_4);
        var currVal_5 = _v.parent.parent.context.$implicit.style;
        _ck(_v, 3, 0, currVal_5);
        var currVal_6 = _v.parent.parent.context.$implicit.icon;
        _ck(_v, 6, 0, currVal_6);
        var currVal_8 = _v.parent.parent.context.$implicit.items;
        _ck(_v, 12, 0, currVal_8);
    }, function (_ck, _v) {
        var currVal_0 = (_v.parent.parent.context.$implicit.url || '#');
        var currVal_1 = _v.parent.parent.context.$implicit.target;
        var currVal_2 = _v.parent.parent.context.$implicit.title;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_7 = _v.parent.parent.context.$implicit.label;
        _ck(_v, 9, 0, currVal_7);
    });
}
function View_MenubarSub_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _v.parent.parent.parent.context.$implicit.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_MenubarSub_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class',
                'ui-submenu-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'fa-caret-down': 0, 'fa-caret-right': 1 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-submenu-icon fa fa-fw';
        var currVal_1 = _ck(_v, 2, 0, _co.root, !_co.root);
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_MenubarSub_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 18, 'a', [], [[1, 'target', 0], [1, 'title', 0], [1, 'target', 0], [8, 'href', 4]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 4).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.itemClick($event, _v.parent.parent.context.$implicit) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-menuitem-link ui-corner-all': 0, 'ui-state-disabled': 1 }),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), i0.ɵdid(671744, [[2, 4]], 0, i3.RouterLinkWithHref, [i3.Router, i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null), i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef, i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'], routerLinkActive: [1,
                'routerLinkActive'] }, null), i0.ɵqud(603979776, 1, { links: 1 }), i0.ɵqud(603979776, 2, { linksWithHrefs: 1 }), i0.ɵpod({ exact: 0 }), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_8)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_9)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_4 = _v.parent.parent.context.$implicit.styleClass;
        var currVal_5 = _ck(_v, 2, 0, true, _v.parent.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_4, currVal_5);
        var currVal_6 = _v.parent.parent.context.$implicit.style;
        _ck(_v, 3, 0, currVal_6);
        var currVal_7 = _v.parent.parent.context.$implicit.routerLink;
        _ck(_v, 4, 0, currVal_7);
        var currVal_8 = (_v.parent.parent.context.$implicit.routerLinkActiveOptions || _ck(_v, 8, 0, false));
        var currVal_9 = 'ui-state-active';
        _ck(_v, 5, 0, currVal_8, currVal_9);
        var currVal_10 = _v.parent.parent.context.$implicit.icon;
        _ck(_v, 11, 0, currVal_10);
        var currVal_12 = _v.parent.parent.context.$implicit.items;
        _ck(_v, 17, 0, currVal_12);
    }, function (_ck, _v) {
        var currVal_0 = _v.parent.parent.context.$implicit.target;
        var currVal_1 = _v.parent.parent.context.$implicit.title;
        var currVal_2 = i0.ɵnov(_v, 4).target;
        var currVal_3 = i0.ɵnov(_v, 4).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
        var currVal_11 = _v.parent.parent.context.$implicit.label;
        _ck(_v, 14, 0, currVal_11);
    });
}
function View_MenubarSub_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-menubarSub', [['class', 'ui-submenu']], null, null, null, View_MenubarSub_0, exports.RenderType_MenubarSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.MenubarSub, [i4.DomHandler], { item: [0,
                'item'] }, null)], function (_ck, _v) {
        var currVal_0 = _v.parent.parent.context.$implicit;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
function View_MenubarSub_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, [['listItem', 1]], null, 12, 'li', [], null, [[null, 'mouseenter'], [null, 'mouseleave']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('mouseenter' === en)) {
                var pd_0 = (_co.onItemMouseEnter($event, i0.ɵnov(_v, 0), _v.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            if (('mouseleave' === en)) {
                var pd_1 = (_co.onItemMouseLeave($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-menuitem ui-widget ui-corner-all': 0, 'ui-menu-parent': 1, 'ui-menuitem-active': 2 }),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_4)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_7)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_10)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _ck(_v, 2, 0, true, _v.parent.context.$implicit.items, (i0.ɵnov(_v, 0) == _co.activeItem));
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = !_v.parent.context.$implicit.routerLink;
        _ck(_v, 5, 0, currVal_1);
        var currVal_2 = _v.parent.context.$implicit.routerLink;
        _ck(_v, 8, 0, currVal_2);
        var currVal_3 = _v.parent.context.$implicit.items;
        _ck(_v, 11, 0, currVal_3);
    }, null);
}
function View_MenubarSub_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.separator;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.separator;
        _ck(_v, 4, 0, currVal_1);
    }, null);
}
function View_MenubarSub_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 11, 'ul', [['class', 'ui-menu-list']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.listClick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-menubar-root-list ui-helper-clearfix': 0,
            'ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-child ui-shadow': 1 }),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MenubarSub_1)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 3, 'li', [['class', 'ui-menuitem ui-menuitem-custom ui-widget ui-corner-all']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-menu-list';
        var currVal_1 = _ck(_v, 3, 0, _co.root, !_co.root);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = (_co.root ? _co.item : _co.item.items);
        _ck(_v, 6, 0, currVal_2);
    }, null);
}
exports.View_MenubarSub_0 = View_MenubarSub_0;
function View_MenubarSub_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-menubarSub', [], null, null, null, View_MenubarSub_0, exports.RenderType_MenubarSub)),
        i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.MenubarSub, [i4.DomHandler], null, null)], null, null);
}
exports.View_MenubarSub_Host_0 = View_MenubarSub_Host_0;
exports.MenubarSubNgFactory = i0.ɵccf('p-menubarSub', i1.MenubarSub, View_MenubarSub_Host_0, { item: 'item', root: 'root' }, {}, ['*']);
