// primeng/components/orderlist/orderlist.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/orderlist/orderlist");
var i2 = require("@angular/common");
var i3 = require("primeng/components/button/button");
var i4 = require("primeng/components/common/shared");
var i5 = require("primeng/components/dom/domhandler");
var i6 = require("primeng/components/utils/objectutils");
exports.OrderListModuleNgFactory = i0.ɵcmf(i1.OrderListModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.ButtonModule, i3.ButtonModule, []), i0.ɵmpd(512, i4.SharedModule, i4.SharedModule, []),
        i0.ɵmpd(512, i1.OrderListModule, i1.OrderListModule, [])]);
});
var styles_OrderList = [];
exports.RenderType_OrderList = i0.ɵcrt({ encapsulation: 2, styles: styles_OrderList,
    data: {} });
function View_OrderList_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['class',
                'ui-orderlist-caption ui-widget-header ui-corner-top']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['',
            '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.header;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_OrderList_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'div', [['class',
                'ui-orderlist-filter-container ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'input', [['class', 'ui-inputtext ui-widget ui-state-default ui-corner-all'],
            ['role', 'textbox'], ['type', 'text']], [[8, 'disabled', 0], [1, 'placeholder', 0]], [[null, 'keyup']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('keyup' === en)) {
                var pd_0 = (_co.onFilterKeyup($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class', 'fa fa-search']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.disabled;
        var currVal_1 = _co.filterPlaceholder;
        _ck(_v, 2, 0, currVal_0, currVal_1);
    });
}
function View_OrderList_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'li', [['class',
                'ui-orderlist-droppoint']], null, [[null, 'dragover'], [null,
                'drop'], [null, 'dragleave']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('dragover' === en)) {
                var pd_0 = (_co.onDragOver($event, _v.parent.context.index) !== false);
                ad = (pd_0 && ad);
            }
            if (('drop' === en)) {
                var pd_1 = (_co.onDrop($event, _v.parent.context.index) !== false);
                ad = (pd_1 && ad);
            }
            if (('dragleave' === en)) {
                var pd_2 = (_co.onDragLeave($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-highlight': 0 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-orderlist-droppoint';
        var currVal_1 = _ck(_v, 2, 0, (_v.parent.context.index === _co.dragOverItemIndex));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_OrderList_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_OrderList_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'li', [['class',
                'ui-orderlist-droppoint']], null, [[null, 'dragover'], [null,
                'drop'], [null, 'dragleave']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('dragover' === en)) {
                var pd_0 = (_co.onDragOver($event, (_v.parent.context.index + 1)) !== false);
                ad = (pd_0 && ad);
            }
            if (('drop' === en)) {
                var pd_1 = (_co.onDrop($event, (_v.parent.context.index + 1)) !== false);
                ad = (pd_1 && ad);
            }
            if (('dragleave' === en)) {
                var pd_2 = (_co.onDragLeave($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-highlight': 0 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-orderlist-droppoint';
        var currVal_1 = _ck(_v, 2, 0, ((_v.parent.context.index + 1) === _co.dragOverItemIndex));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_OrderList_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrderList_4)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 6, 'li', [['class', 'ui-orderlist-item']], [[4, 'display', null], [8, 'draggable', 0]], [[null, 'click'],
            [null, 'touchend'], [null, 'dragstart'], [null,
                'dragend']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onItemClick($event, _v.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            if (('touchend' === en)) {
                var pd_1 = (_co.onItemTouchEnd($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('dragstart' === en)) {
                var pd_2 = (_co.onDragStart($event, _v.context.index) !== false);
                ad = (pd_2 && ad);
            }
            if (('dragend' === en)) {
                var pd_3 = (_co.onDragEnd($event) !== false);
                ad = (pd_3 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-state-highlight': 0 }),
        (_l()(), i0.ɵted(null, ['\n                            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrderList_5)), i0.ɵdid(212992, null, 0, i4.TemplateWrapper, [i0.ViewContainerRef], { item: [0, 'item'],
            templateRef: [1, 'templateRef'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrderList_6)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.dragdrop && _co.isItemVisible(_v.context.$implicit));
        _ck(_v, 2, 0, currVal_0);
        var currVal_3 = 'ui-orderlist-item';
        var currVal_4 = _ck(_v, 6, 0, _co.isSelected(_v.context.$implicit));
        _ck(_v, 5, 0, currVal_3, currVal_4);
        var currVal_5 = _v.context.$implicit;
        var currVal_6 = _co.itemTemplate;
        _ck(_v, 9, 0, currVal_5, currVal_6);
        var currVal_7 = (_co.dragdrop && _v.context.last);
        _ck(_v, 13, 0, currVal_7);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = (_co.isItemVisible(_v.context.$implicit) ? 'block' : 'none');
        var currVal_2 = _co.dragdrop;
        _ck(_v, 4, 0, currVal_1, currVal_2);
    });
}
function View_OrderList_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { listViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 39, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-orderlist ui-widget': 0,
            'ui-orderlist-responsive': 1 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 17, 'div', [['class', 'ui-orderlist-controls']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, null, null, 2, 'button', [['icon', 'fa-angle-up'],
            ['pButton', ''], ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.moveUp($event, i0.ɵnov(_v, 34)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []), i0.ɵdid(4341760, null, 0, i3.Button, [i0.ElementRef,
            i5.DomHandler], { icon: [0, 'icon'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 2, 'button', [['icon', 'fa-angle-double-up'], ['pButton', ''], ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.moveTop($event, i0.ɵnov(_v, 34)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []), i0.ɵdid(4341760, null, 0, i3.Button, [i0.ElementRef,
            i5.DomHandler], { icon: [0, 'icon'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 2, 'button', [['icon', 'fa-angle-down'], ['pButton', ''], ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.moveDown($event, i0.ɵnov(_v, 34)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []), i0.ɵdid(4341760, null, 0, i3.Button, [i0.ElementRef,
            i5.DomHandler], { icon: [0, 'icon'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 2, 'button', [['icon', 'fa-angle-double-down'], ['pButton', ''], ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.moveBottom($event, i0.ɵnov(_v, 34)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []), i0.ɵdid(4341760, null, 0, i3.Button, [i0.ElementRef,
            i5.DomHandler], { icon: [0, 'icon'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 14, 'div', [['class', 'ui-orderlist-list-container']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrderList_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_OrderList_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵeld(0, [[1, 0], ['listelement', 1]], null, 5, 'ul', [['class', 'ui-widget-content ui-orderlist-list ui-corner-bottom']], null, [[null, 'dragover']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('dragover' === en)) {
                var pd_0 = (_co.onListMouseMove($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_OrderList_3)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 4, 0, true, _co.responsive);
        _ck(_v, 3, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 5, 0, currVal_2);
        var currVal_3 = 'fa-angle-up';
        _ck(_v, 11, 0, currVal_3);
        var currVal_4 = 'fa-angle-double-up';
        _ck(_v, 15, 0, currVal_4);
        var currVal_5 = 'fa-angle-down';
        _ck(_v, 19, 0, currVal_5);
        var currVal_6 = 'fa-angle-double-down';
        _ck(_v, 23, 0, currVal_6);
        var currVal_7 = _co.header;
        _ck(_v, 29, 0, currVal_7);
        var currVal_8 = _co.filterBy;
        _ck(_v, 32, 0, currVal_8);
        var currVal_9 = _co.listStyle;
        _ck(_v, 35, 0, currVal_9);
        var currVal_10 = _co.value;
        _ck(_v, 38, 0, currVal_10);
    }, null);
}
exports.View_OrderList_0 = View_OrderList_0;
function View_OrderList_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'p-orderList', [], null, null, null, View_OrderList_0, exports.RenderType_OrderList)),
        i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []), i0.ɵprd(512, null, i6.ObjectUtils, i6.ObjectUtils, []), i0.ɵdid(13680640, null, 1, i1.OrderList, [i0.ElementRef, i5.DomHandler, i6.ObjectUtils], null, null), i0.ɵqud(603979776, 1, { templates: 1 })], null, null);
}
exports.View_OrderList_Host_0 = View_OrderList_Host_0;
exports.OrderListNgFactory = i0.ɵccf('p-orderList', i1.OrderList, View_OrderList_Host_0, { header: 'header', style: 'style', styleClass: 'styleClass',
    listStyle: 'listStyle', responsive: 'responsive', filterBy: 'filterBy', filterPlaceholder: 'filterPlaceholder',
    metaKeySelection: 'metaKeySelection', dragdrop: 'dragdrop', dragdropScope: 'dragdropScope',
    value: 'value' }, { onReorder: 'onReorder', onSelectionChange: 'onSelectionChange',
    onFilterEvent: 'onFilterEvent' }, []);
