// primeng/components/tabmenu/tabmenu.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/tabmenu/tabmenu");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
var i4 = require("primeng/components/dom/domhandler");
exports.TabMenuModuleNgFactory = i0.ɵcmf(i1.TabMenuModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.RouterModule, i3.RouterModule, [[2, i3.ɵa], [2, i3.Router]]), i0.ɵmpd(512, i1.TabMenuModule, i1.TabMenuModule, [])]);
});
var styles_TabMenu = [];
exports.RenderType_TabMenu = i0.ɵcrt({ encapsulation: 2, styles: styles_TabMenu,
    data: {} });
function View_TabMenu_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 7, 'a', [['class',
                'ui-menuitem-link ui-corner-all']], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.itemClick($event, _v.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-icon fa']], null, null, null, null, null)),
        i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_3 = 'ui-menuitem-icon fa';
        var currVal_4 = _v.parent.context.$implicit.icon;
        _ck(_v, 3, 0, currVal_3, currVal_4);
    }, function (_ck, _v) {
        var currVal_0 = (_v.parent.context.$implicit.url || '#');
        var currVal_1 = _v.parent.context.$implicit.target;
        var currVal_2 = _v.parent.context.$implicit.title;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_5 = _v.parent.context.$implicit.label;
        _ck(_v, 6, 0, currVal_5);
    });
}
function View_TabMenu_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 12, 'a', [['class',
                'ui-menuitem-link ui-corner-all']], [[1, 'target', 0], [1, 'title', 0], [1, 'target',
                0], [8, 'href', 4]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.itemClick($event, _v.parent.context.$implicit) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(671744, [[2, 4]], 0, i3.RouterLinkWithHref, [i3.Router,
            i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null),
        i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef,
            i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'],
            routerLinkActive: [1, 'routerLinkActive'] }, null), i0.ɵqud(603979776, 1, { links: 1 }), i0.ɵqud(603979776, 2, { linksWithHrefs: 1 }), i0.ɵpod({ exact: 0 }),
        (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-icon fa']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_4 = _v.parent.context.$implicit.routerLink;
        _ck(_v, 1, 0, currVal_4);
        var currVal_5 = (_v.parent.context.$implicit.routerLinkActiveOptions || _ck(_v, 5, 0, false));
        var currVal_6 = 'ui-state-active';
        _ck(_v, 2, 0, currVal_5, currVal_6);
        var currVal_7 = 'ui-menuitem-icon fa';
        var currVal_8 = _v.parent.context.$implicit.icon;
        _ck(_v, 8, 0, currVal_7, currVal_8);
    }, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.target;
        var currVal_1 = _v.parent.context.$implicit.title;
        var currVal_2 = i0.ɵnov(_v, 1).target;
        var currVal_3 = i0.ɵnov(_v, 1).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
        var currVal_9 = _v.parent.context.$implicit.label;
        _ck(_v, 11, 0, currVal_9);
    });
}
function View_TabMenu_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 9, 'li', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngClass: [0, 'ngClass'] }, null), i0.ɵpod({ 'ui-tabmenuitem ui-state-default ui-corner-top': 0,
            'ui-state-disabled': 1, 'ui-tabmenuitem-hasicon': 2, 'ui-state-active': 3 }), (_l()(),
            i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TabMenu_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TabMenu_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _ck(_v, 2, 0, true, _v.context.$implicit.disabled, _v.context.$implicit.icon, (_co.activeItem == _v.context.$implicit));
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.routerLink;
        _ck(_v, 5, 0, currVal_1);
        var currVal_2 = _v.context.$implicit.routerLink;
        _ck(_v, 8, 0, currVal_2);
    }, null);
}
function View_TabMenu_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 9, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 4, 'ul', [['class', 'ui-tabmenu-nav ui-helper-reset ui-helper-clearfix ui-widget-header ui-corner-all'],
            ['role', 'tablist']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_TabMenu_1)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-tabmenu ui-widget ui-widget-content ui-corner-all';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = _co.model;
        _ck(_v, 8, 0, currVal_3);
    }, null);
}
exports.View_TabMenu_0 = View_TabMenu_0;
function View_TabMenu_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-tabMenu', [], null, null, null, View_TabMenu_0, exports.RenderType_TabMenu)),
        i0.ɵprd(4608, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(114688, null, 0, i1.TabMenu, [], null, null)], function (_ck, _v) {
        _ck(_v, 2, 0);
    }, null);
}
exports.View_TabMenu_Host_0 = View_TabMenu_Host_0;
exports.TabMenuNgFactory = i0.ɵccf('p-tabMenu', i1.TabMenu, View_TabMenu_Host_0, { model: 'model', activeItem: 'activeItem', popup: 'popup',
    style: 'style', styleClass: 'styleClass' }, {}, []);
