// primeng/components/inputmask/inputmask.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/inputmask/inputmask");
var i2 = require("@angular/common");
var i3 = require("primeng/components/inputtext/inputtext");
var i4 = require("@angular/forms");
var i5 = require("primeng/components/dom/domhandler");
exports.InputMaskModuleNgFactory = i0.ɵcmf(i1.InputMaskModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.InputTextModule, i3.InputTextModule, []), i0.ɵmpd(512, i1.InputMaskModule, i1.InputMaskModule, [])]);
});
var styles_InputMask = [];
exports.RenderType_InputMask = i0.ɵcrt({ encapsulation: 2, styles: styles_InputMask,
    data: {} });
function View_InputMask_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { inputViewChild: 0 }), (_l()(), i0.ɵeld(0, [[1,
                0], ['input', 1]], null, 3, 'input', [['pInputText', '']], [[1, 'id', 0], [1, 'type',
                0], [1, 'name', 0], [1, 'placeholder', 0], [1, 'size', 0], [1, 'maxlength', 0], [1, 'tabindex',
                0], [8, 'disabled', 0], [8, 'readOnly', 0], [1, 'required', 0], [2, 'ui-inputtext', null],
            [2, 'ui-corner-all', null], [2, 'ui-state-default', null], [2, 'ui-widget',
                null], [2, 'ui-state-filled', null]], [[null, 'focus'],
            [null, 'blur'], [null, 'keydown'], [null, 'keypress'],
            [null, 'input'], [null, 'paste']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('input' === en)) {
                var pd_0 = (i0.ɵnov(_v, 4).onInput($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('focus' === en)) {
                var pd_1 = (_co.onInputFocus($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('blur' === en)) {
                var pd_2 = (_co.onInputBlur($event) !== false);
                ad = (pd_2 && ad);
            }
            if (('keydown' === en)) {
                var pd_3 = (_co.onKeyDown($event) !== false);
                ad = (pd_3 && ad);
            }
            if (('keypress' === en)) {
                var pd_4 = (_co.onKeyPress($event) !== false);
                ad = (pd_4 && ad);
            }
            if (('input' === en)) {
                var pd_5 = (_co.onInput($event) !== false);
                ad = (pd_5 && ad);
            }
            if (('paste' === en)) {
                var pd_6 = (_co.handleInputChange($event) !== false);
                ad = (pd_6 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0, 'ngClass'] }, null),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), i0.ɵdid(278528, null, 0, i3.InputText, [i0.ElementRef], null, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_15 = _co.styleClass;
        _ck(_v, 2, 0, currVal_15);
        var currVal_16 = _co.style;
        _ck(_v, 3, 0, currVal_16);
        _ck(_v, 4, 0);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.inputId;
        var currVal_1 = _co.type;
        var currVal_2 = _co.name;
        var currVal_3 = _co.placeholder;
        var currVal_4 = _co.size;
        var currVal_5 = _co.maxlength;
        var currVal_6 = _co.tabindex;
        var currVal_7 = _co.disabled;
        var currVal_8 = _co.readonly;
        var currVal_9 = _co.required;
        var currVal_10 = true;
        var currVal_11 = true;
        var currVal_12 = true;
        var currVal_13 = true;
        var currVal_14 = i0.ɵnov(_v, 4).filled;
        _ck(_v, 1, 1, [currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6,
            currVal_7, currVal_8, currVal_9, currVal_10, currVal_11, currVal_12, currVal_13,
            currVal_14]);
    });
}
exports.View_InputMask_0 = View_InputMask_0;
function View_InputMask_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'p-inputMask', [], [[2, 'ui-inputwrapper-filled', null], [2, 'ui-inputwrapper-focus',
                null]], null, null, View_InputMask_0, exports.RenderType_InputMask)),
        i0.ɵprd(5120, null, i4.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i1.InputMask]), i0.ɵprd(512, null, i5.DomHandler, i5.DomHandler, []),
        i0.ɵdid(245760, null, 0, i1.InputMask, [i0.ElementRef, i5.DomHandler], null, null)], function (_ck, _v) {
        _ck(_v, 3, 0);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 3).filled;
        var currVal_1 = i0.ɵnov(_v, 3).focus;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
exports.View_InputMask_Host_0 = View_InputMask_Host_0;
exports.InputMaskNgFactory = i0.ɵccf('p-inputMask', i1.InputMask, View_InputMask_Host_0, { type: 'type', slotChar: 'slotChar', autoClear: 'autoClear',
    style: 'style', inputId: 'inputId', styleClass: 'styleClass', placeholder: 'placeholder',
    size: 'size', maxlength: 'maxlength', tabindex: 'tabindex', disabled: 'disabled',
    readonly: 'readonly', unmask: 'unmask', name: 'name', required: 'required', mask: 'mask' }, { onComplete: 'onComplete', onFocus: 'onFocus', onBlur: 'onBlur' }, []);
