// primeng/components/accordion/accordion.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/accordion/accordion");
var i2 = require("@angular/common");
exports.AccordionModuleNgFactory = i0.ɵcmf(i1.AccordionModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.AccordionModule, i1.AccordionModule, [])]);
});
var styles_Accordion = [];
exports.RenderType_Accordion = i0.ɵcrt({ encapsulation: 2, styles: styles_Accordion,
    data: {} });
function View_Accordion_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 5, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-accordion ui-widget ui-helper-reset';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
    }, null);
}
exports.View_Accordion_0 = View_Accordion_0;
function View_Accordion_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-accordion', [], null, null, null, View_Accordion_0, exports.RenderType_Accordion)),
        i0.ɵdid(49152, null, 0, i1.Accordion, [i0.ElementRef], null, null)], null, null);
}
exports.View_Accordion_Host_0 = View_Accordion_Host_0;
exports.AccordionNgFactory = i0.ɵccf('p-accordion', i1.Accordion, View_Accordion_Host_0, { multiple: 'multiple', style: 'style', styleClass: 'styleClass',
    lazy: 'lazy', activeIndex: 'activeIndex' }, { onClose: 'onClose', onOpen: 'onOpen' }, ['*']);
var styles_AccordionTab = [];
exports.RenderType_AccordionTab = i0.ɵcrt({ encapsulation: 2, styles: styles_AccordionTab,
    data: { 'animation': [{ type: 7, name: 'tabContent', definitions: [{ type: 0, name: 'hidden',
                        styles: { type: 6, styles: { height: '0' }, offset: null }, options: undefined },
                    { type: 0, name: 'visible', styles: { type: 6, styles: { height: '*' }, offset: null },
                        options: undefined }, { type: 1, expr: 'visible <=> hidden', animation: { type: 4,
                            styles: null, timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)' }, options: null }],
                options: {} }] } });
function View_AccordionTab_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'a', [['href', '#'],
            ['role', 'tab']], [[1, 'aria-expanded', 0], [1, 'aria-selected', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['',
            '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.selected;
        var currVal_1 = _co.selected;
        _ck(_v, 0, 0, currVal_0, currVal_1);
        var currVal_2 = _co.header;
        _ck(_v, 1, 0, currVal_2);
    });
}
function View_AccordionTab_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'a', [['href', '#'],
            ['role', 'tab']], [[1, 'aria-expanded', 0], [1, 'aria-selected', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.selected;
        var currVal_1 = _co.selected;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
function View_AccordionTab_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-accordion-content ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n            ']))], null, null);
}
function View_AccordionTab_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 13, 'div', [['class', 'ui-accordion-header ui-state-default ui-corner-all']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggle($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-active': 0, 'ui-state-disabled': 1 }),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class', 'fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'fa-caret-down': 0,
            'fa-caret-right': 1 }), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_AccordionTab_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_AccordionTab_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'div', [['class', 'ui-accordion-content-wrapper'], ['role', 'tabpanel']], [[24,
                '@tabContent', 0], [1, 'aria-hidden', 0]], [[null, '@tabContent.done']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('@tabContent.done' === en)) {
                var pd_0 = (_co.onToggleDone($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-accordion-content-wrapper-overflown': 0 }),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_AccordionTab_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-accordion-header ui-state-default ui-corner-all';
        var currVal_1 = _ck(_v, 3, 0, _co.selected, _co.disabled);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = 'fa fa-fw';
        var currVal_3 = _ck(_v, 7, 0, _co.selected, !_co.selected);
        _ck(_v, 6, 0, currVal_2, currVal_3);
        var currVal_4 = !_co.hasHeaderFacet;
        _ck(_v, 10, 0, currVal_4);
        var currVal_5 = _co.hasHeaderFacet;
        _ck(_v, 13, 0, currVal_5);
        var currVal_8 = 'ui-accordion-content-wrapper';
        var currVal_9 = _ck(_v, 18, 0, (!_co.selected || _co.animating));
        _ck(_v, 17, 0, currVal_8, currVal_9);
        var currVal_10 = (_co.lazy ? _co.selected : true);
        _ck(_v, 21, 0, currVal_10);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_6 = (_co.selected ? 'visible' : 'hidden');
        var currVal_7 = !_co.selected;
        _ck(_v, 16, 0, currVal_6, currVal_7);
    });
}
exports.View_AccordionTab_0 = View_AccordionTab_0;
function View_AccordionTab_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-accordionTab', [], null, null, null, View_AccordionTab_0, exports.RenderType_AccordionTab)), i0.ɵdid(180224, null, 1, i1.AccordionTab, [i1.Accordion], null, null), i0.ɵqud(603979776, 1, { headerFacet: 1 })], null, null);
}
exports.View_AccordionTab_Host_0 = View_AccordionTab_Host_0;
exports.AccordionTabNgFactory = i0.ɵccf('p-accordionTab', i1.AccordionTab, View_AccordionTab_Host_0, { header: 'header', selected: 'selected',
    disabled: 'disabled' }, { selectedChange: 'selectedChange' }, ['p-header', '*']);
