// primeng/components/tristatecheckbox/tristatecheckbox.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/tristatecheckbox/tristatecheckbox");
var i2 = require("@angular/common");
var i3 = require("@angular/forms");
exports.TriStateCheckboxModuleNgFactory = i0.ɵcmf(i1.TriStateCheckboxModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.TriStateCheckboxModule, i1.TriStateCheckboxModule, [])]);
});
var styles_TriStateCheckbox = [];
exports.RenderType_TriStateCheckbox = i0.ɵcrt({ encapsulation: 2,
    styles: styles_TriStateCheckbox, data: {} });
function View_TriStateCheckbox_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 17, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'ui-helper-hidden-accessible']], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, [['input', 1]], null, 0, 'input', [['readonly', ''], ['type', 'text']], [[1, 'id', 0], [8, 'name',
                0], [1, 'tabindex', 0], [8, 'disabled', 0]], [[null, 'keyup'], [null,
                'keydown'], [null, 'focus'], [null, 'blur']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('keyup' === en)) {
                var pd_0 = (_co.onKeyup($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('keydown' === en)) {
                var pd_1 = (_co.onKeydown($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('focus' === en)) {
                var pd_2 = (_co.onFocus() !== false);
                ad = (pd_2 && ad);
            }
            if (('blur' === en)) {
                var pd_3 = (_co.onBlur() !== false);
                ad = (pd_3 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 7, 'div', [['class', 'ui-chkbox-box ui-widget ui-corner-all ui-state-default']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onClick($event, i0.ɵnov(_v, 7)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-state-active': 0,
            'ui-state-disabled': 1, 'ui-state-focus': 2 }), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class', 'ui-chkbox-icon fa ui-clickable']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'fa-check': 0,
            'fa-close': 1 }), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(),
            i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-chkbox ui-tristatechkbox ui-widget';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_7 = 'ui-chkbox-box ui-widget ui-corner-all ui-state-default';
        var currVal_8 = _ck(_v, 12, 0, (_co.value != null), _co.disabled, _co.focus);
        _ck(_v, 11, 0, currVal_7, currVal_8);
        var currVal_9 = 'ui-chkbox-icon fa ui-clickable';
        var currVal_10 = _ck(_v, 16, 0, (_co.value == true), (_co.value == false));
        _ck(_v, 15, 0, currVal_9, currVal_10);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_3 = _co.inputId;
        var currVal_4 = _co.name;
        var currVal_5 = _co.tabindex;
        var currVal_6 = _co.disabled;
        _ck(_v, 7, 0, currVal_3, currVal_4, currVal_5, currVal_6);
    });
}
exports.View_TriStateCheckbox_0 = View_TriStateCheckbox_0;
function View_TriStateCheckbox_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-triStateCheckbox', [], null, null, null, View_TriStateCheckbox_0, exports.RenderType_TriStateCheckbox)), i0.ɵprd(5120, null, i3.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i1.TriStateCheckbox]), i0.ɵdid(49152, null, 0, i1.TriStateCheckbox, [i0.ChangeDetectorRef], null, null)], null, null);
}
exports.View_TriStateCheckbox_Host_0 = View_TriStateCheckbox_Host_0;
exports.TriStateCheckboxNgFactory = i0.ɵccf('p-triStateCheckbox', i1.TriStateCheckbox, View_TriStateCheckbox_Host_0, { disabled: 'disabled', name: 'name',
    tabindex: 'tabindex', inputId: 'inputId', style: 'style', styleClass: 'styleClass' }, { onChange: 'onChange' }, []);
