// @ng-bootstrap/ng-bootstrap/carousel/carousel.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@angular/common");
var i2 = require("@ng-bootstrap/ng-bootstrap/carousel/carousel");
var i3 = require("@ng-bootstrap/ng-bootstrap/carousel/carousel-config");
var styles_NgbCarousel = [];
exports.RenderType_NgbCarousel = i0.ɵcrt({ encapsulation: 2, styles: styles_NgbCarousel,
    data: {} });
function View_NgbCarousel_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'li', [], [[8, 'id', 0], [2, 'active', null]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.cycleToSelected(_v.context.$implicit.id, _co._getSlideEventDirection(_co.activeId, _v.context.$implicit.id)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit.id;
        var currVal_1 = (_v.context.$implicit.id === _co.activeId);
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
function View_NgbCarousel_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_NgbCarousel_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'div', [['class',
                'carousel-item']], [[2, 'active', null]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbCarousel_3)), i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n      ']))], function (_ck, _v) {
        var currVal_1 = _v.context.$implicit.tplRef;
        _ck(_v, 3, 0, currVal_1);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_v.context.$implicit.id === _co.activeId);
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_NgbCarousel_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 4, 'ol', [['class', 'carousel-indicators']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n      '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbCarousel_1)),
        i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 4, 'div', [['class', 'carousel-inner']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbCarousel_2)), i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef,
            i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null),
        (_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵted(null, ['\n    '])),
        (_l()(), i0.ɵeld(0, null, null, 6, 'a', [['class', 'carousel-control-prev'],
            ['role', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.cycleToPrev() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n      '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['aria-hidden', 'true'],
            ['class', 'carousel-control-prev-icon']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n      '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'sr-only']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['Previous'])), (_l()(), i0.ɵted(null, ['\n    '])),
        (_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 6, 'a', [['class', 'carousel-control-next'], ['role', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.cycleToNext() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n      '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [['aria-hidden', 'true'],
            ['class', 'carousel-control-next-icon']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n      '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'sr-only']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['Next'])), (_l()(), i0.ɵted(null, ['\n    '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.slides;
        _ck(_v, 4, 0, currVal_0);
        var currVal_1 = _co.slides;
        _ck(_v, 10, 0, currVal_1);
    }, null);
}
exports.View_NgbCarousel_0 = View_NgbCarousel_0;
function View_NgbCarousel_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'ngb-carousel', [['class', 'carousel slide'], ['tabIndex', '0']], [[4, 'display', null]], [[null, 'mouseenter'], [null, 'mouseleave'], [null, 'keydown.arrowLeft'],
            [null, 'keydown.arrowRight']], function (_v, en, $event) {
            var ad = true;
            if (('mouseenter' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1).pause() !== false);
                ad = (pd_0 && ad);
            }
            if (('mouseleave' === en)) {
                var pd_1 = (i0.ɵnov(_v, 1).cycle() !== false);
                ad = (pd_1 && ad);
            }
            if (('keydown.arrowLeft' === en)) {
                var pd_2 = (i0.ɵnov(_v, 1).keyPrev() !== false);
                ad = (pd_2 && ad);
            }
            if (('keydown.arrowRight' === en)) {
                var pd_3 = (i0.ɵnov(_v, 1).keyNext() !== false);
                ad = (pd_3 && ad);
            }
            return ad;
        }, View_NgbCarousel_0, exports.RenderType_NgbCarousel)), i0.ɵdid(2867200, null, 1, i2.NgbCarousel, [i3.NgbCarouselConfig], null, null), i0.ɵqud(603979776, 1, { slides: 1 })], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, function (_ck, _v) {
        var currVal_0 = 'block';
        _ck(_v, 0, 0, currVal_0);
    });
}
exports.View_NgbCarousel_Host_0 = View_NgbCarousel_Host_0;
exports.NgbCarouselNgFactory = i0.ɵccf('ngb-carousel', i2.NgbCarousel, View_NgbCarousel_Host_0, { interval: 'interval', wrap: 'wrap', keyboard: 'keyboard',
    activeId: 'activeId' }, { slide: 'slide' }, []);
