// primeng/components/steps/steps.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/steps/steps");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
exports.StepsModuleNgFactory = i0.ɵcmf(i1.StepsModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.RouterModule, i3.RouterModule, [[2, i3.ɵa], [2, i3.Router]]), i0.ɵmpd(512, i1.StepsModule, i1.StepsModule, [])]);
});
var styles_Steps = [];
exports.RenderType_Steps = i0.ɵcrt({ encapsulation: 2, styles: styles_Steps,
    data: {} });
function View_Steps_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 7, 'a', [['class',
                'ui-menuitem-link']], [[8, 'href', 4], [1, 'target', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.itemClick($event, _v.parent.context.$implicit, _v.parent.context.index) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-steps-number']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-steps-title']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], null, function (_ck, _v) {
        var currVal_0 = (_v.parent.context.$implicit.url || '#');
        var currVal_1 = _v.parent.context.$implicit.target;
        _ck(_v, 0, 0, currVal_0, currVal_1);
        var currVal_2 = (_v.parent.context.index + 1);
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = _v.parent.context.$implicit.label;
        _ck(_v, 6, 0, currVal_3);
    });
}
function View_Steps_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 12, 'a', [['class',
                'ui-menuitem-link']], [[1, 'target', 0], [1, 'target', 0], [8, 'href', 4]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.itemClick($event, _v.parent.context.$implicit, _v.parent.context.index) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(671744, [[2, 4]], 0, i3.RouterLinkWithHref, [i3.Router,
            i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null),
        i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef,
            i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'],
            routerLinkActive: [1, 'routerLinkActive'] }, null), i0.ɵqud(603979776, 1, { links: 1 }), i0.ɵqud(603979776, 2, { linksWithHrefs: 1 }), i0.ɵpod({ exact: 0 }),
        (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-steps-number']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-steps-title']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_3 = _v.parent.context.$implicit.routerLink;
        _ck(_v, 1, 0, currVal_3);
        var currVal_4 = (_v.parent.context.$implicit.routerLinkActiveOptions || _ck(_v, 5, 0, false));
        var currVal_5 = 'ui-state-active';
        _ck(_v, 2, 0, currVal_4, currVal_5);
    }, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.target;
        var currVal_1 = i0.ɵnov(_v, 1).target;
        var currVal_2 = i0.ɵnov(_v, 1).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_6 = (_v.parent.context.index + 1);
        _ck(_v, 8, 0, currVal_6);
        var currVal_7 = _v.parent.context.$implicit.label;
        _ck(_v, 11, 0, currVal_7);
    });
}
function View_Steps_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, [['menuitem', 1]], null, 9, 'li', [['class',
                'ui-steps-item']], null, null, null, null, null)),
        i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-state-highlight': 0, 'ui-state-default': 1, 'ui-state-disabled': 2 }),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Steps_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Steps_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-steps-item';
        var currVal_1 = _ck(_v, 2, 0, (_v.context.index === _co.activeIndex), (_v.context.index !== _co.activeIndex), (_v.context.$implicit.disabled || ((_v.context.index !== _co.activeIndex) && _co.readonly)));
        _ck(_v, 1, 0, currVal_0, currVal_1);
        var currVal_2 = !_v.context.$implicit.routerLink;
        _ck(_v, 5, 0, currVal_2);
        var currVal_3 = _v.context.$implicit.routerLink;
        _ck(_v, 8, 0, currVal_3);
    }, null);
}
function View_Steps_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 10, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-steps ui-widget ui-helper-clearfix': 0,
            'ui-steps-readonly': 1 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 4, 'ul', [['role',
                'tablist']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Steps_1)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 3, 0, true, _co.readonly);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 4, 0, currVal_2);
        var currVal_3 = _co.model;
        _ck(_v, 9, 0, currVal_3);
    }, null);
}
exports.View_Steps_0 = View_Steps_0;
function View_Steps_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-steps', [], null, null, null, View_Steps_0, exports.RenderType_Steps)), i0.ɵdid(49152, null, 0, i1.Steps, [], null, null)], null, null);
}
exports.View_Steps_Host_0 = View_Steps_Host_0;
exports.StepsNgFactory = i0.ɵccf('p-steps', i1.Steps, View_Steps_Host_0, { activeIndex: 'activeIndex', model: 'model', readonly: 'readonly',
    style: 'style', styleClass: 'styleClass' }, { activeIndexChange: 'activeIndexChange' }, []);
