// @ng-bootstrap/ng-bootstrap/typeahead/highlight.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@angular/common");
var i2 = require("@ng-bootstrap/ng-bootstrap/typeahead/highlight");
var styles_NgbHighlight = ['.ngb-highlight[_ngcontent-%COMP%] {\n      font-weight: bold;\n    }'];
exports.RenderType_NgbHighlight = i0.ɵcrt({ encapsulation: 0, styles: styles_NgbHighlight,
    data: {} });
function View_NgbHighlight_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [], [[8, 'className', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = i0.ɵinlineInterpolate(1, '', _co.highlightClass, '');
        _ck(_v, 0, 0, currVal_0);
        var currVal_1 = _v.parent.context.$implicit;
        _ck(_v, 1, 0, currVal_1);
    });
}
function View_NgbHighlight_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit;
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_NgbHighlight_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbHighlight_2)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbHighlight_3)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵand(0, null, null, 0))], function (_ck, _v) {
        var currVal_0 = _v.context.odd;
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = !_v.context.odd;
        _ck(_v, 3, 0, currVal_1);
    }, null);
}
function View_NgbHighlight_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbHighlight_1)), i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef,
            i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.parts;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
exports.View_NgbHighlight_0 = View_NgbHighlight_0;
function View_NgbHighlight_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'ngb-highlight', [], null, null, null, View_NgbHighlight_0, exports.RenderType_NgbHighlight)), i0.ɵdid(573440, null, 0, i2.NgbHighlight, [], null, null)], null, null);
}
exports.View_NgbHighlight_Host_0 = View_NgbHighlight_Host_0;
exports.NgbHighlightNgFactory = i0.ɵccf('ngb-highlight', i2.NgbHighlight, View_NgbHighlight_Host_0, { highlightClass: 'highlightClass', result: 'result',
    term: 'term' }, {}, []);
