// primeng/components/carousel/carousel.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/carousel/carousel");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
var i4 = require("primeng/components/dom/domhandler");
exports.CarouselModuleNgFactory = i0.ɵcmf(i1.CarouselModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i1.CarouselModule, i1.CarouselModule, [])]);
});
var styles_Carousel = [];
exports.RenderType_Carousel = i0.ɵcrt({ encapsulation: 2, styles: styles_Carousel,
    data: {} });
function View_Carousel_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class',
                'ui-carousel-button ui-carousel-next-button fa fa-arrow-circle-right']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onNextNav() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-carousel-button ui-carousel-next-button fa fa-arrow-circle-right';
        var currVal_1 = _ck(_v, 2, 0, ((_co.page === (_co.totalPages - 1)) && !_co.circular));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Carousel_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class',
                'ui-carousel-button ui-carousel-prev-button fa fa-arrow-circle-left']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onPrevNav() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-state-disabled': 0 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-carousel-button ui-carousel-prev-button fa fa-arrow-circle-left';
        var currVal_1 = _ck(_v, 2, 0, ((_co.page === 0) && !_co.circular));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Carousel_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'a', [['class',
                'ui-carousel-page-link fa fa-circle-o'], ['href', '#']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.setPageWithLink($event, _v.context.index) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'fa-dot-circle-o': 0 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-carousel-page-link fa fa-circle-o';
        var currVal_1 = _ck(_v, 2, 0, (_co.page === _v.context.index));
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Carousel_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'div', [['class',
                'ui-carousel-page-links']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Carousel_4)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.anchorPageLinks;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_Carousel_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'option', [], [[8, 'value', 0], [8, 'selected', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit;
        var currVal_1 = (_co.value == _v.context.$implicit);
        _ck(_v, 0, 0, currVal_0, currVal_1);
        var currVal_2 = (_v.context.$implicit + 1);
        _ck(_v, 1, 0, currVal_2);
    });
}
function View_Carousel_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'select', [['class',
                'ui-carousel-dropdown ui-widget ui-state-default ui-corner-left']], [[8, 'value',
                0]], [[null, 'change']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('change' === en)) {
                var pd_0 = (_co.onDropdownChange($event.target.value) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_6)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _co.selectDropdownOptions;
        _ck(_v, 3, 0, currVal_1);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.page;
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_Carousel_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'option', [], [[8, 'value', 0], [8, 'selected', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit;
        var currVal_1 = (_co.value == _v.context.$implicit);
        _ck(_v, 0, 0, currVal_0, currVal_1);
        var currVal_2 = (_v.context.$implicit + 1);
        _ck(_v, 1, 0, currVal_2);
    });
}
function View_Carousel_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'select', [['class',
                'ui-carousel-mobiledropdown ui-widget ui-state-default ui-corner-left']], [[8,
                'value', 0], [4, 'display', null]], [[null, 'change']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('change' === en)) {
                var pd_0 = (_co.onDropdownChange($event.target.value) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_8)),
        i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = _co.mobileDropdownOptions;
        _ck(_v, 3, 0, currVal_2);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.page;
        var currVal_1 = (_co.shrinked ? 'block' : 'none');
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
function View_Carousel_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_Carousel_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'li', [['class',
                'ui-carousel-item ui-widget-content ui-corner-all']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_10)),
        i0.ɵdid(212992, null, 0, i3.TemplateWrapper, [i0.ViewContainerRef], { item: [0,
                'item'], templateRef: [1, 'templateRef'] }, null), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _v.context.$implicit;
        var currVal_1 = _co.itemTemplate;
        _ck(_v, 3, 0, currVal_0, currVal_1);
    }, null);
}
function View_Carousel_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { containerViewChild: 0 }), i0.ɵqud(402653184, 2, { viewportViewChild: 0 }), i0.ɵqud(402653184, 3, { itemsViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, [[1, 0], ['container', 1]], null, 34, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-carousel ui-widget ui-widget-content ui-corner-all': 0 }),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 19, 'div', [['class', 'ui-carousel-header ui-widget-header ui-corner-all']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-carousel-header-title']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Carousel_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_5)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_7)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, [[2, 0],
            ['viewport', 1]], null, 7, 'div', [['class', 'ui-carousel-viewport']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, [[3,
                0], ['items', 1]], null, 4, 'ul', [['class', 'ui-carousel-items']], [[4,
                'left', 'px'], [4, 'transitionProperty', null], [4, 'transitionDuration',
                null], [4, 'transitionTimingFunction', null]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Carousel_9)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 6, 0, true);
        _ck(_v, 5, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 7, 0, currVal_2);
        var currVal_4 = (_co.value && _co.value.length);
        _ck(_v, 15, 0, currVal_4);
        var currVal_5 = (_co.value && _co.value.length);
        _ck(_v, 18, 0, currVal_5);
        var currVal_6 = _co.displayPageLinks;
        _ck(_v, 21, 0, currVal_6);
        var currVal_7 = _co.displayPageDropdown;
        _ck(_v, 24, 0, currVal_7);
        var currVal_8 = ((_co.responsive && _co.value) && _co.value.length);
        _ck(_v, 27, 0, currVal_8);
        var currVal_13 = _co.value;
        _ck(_v, 35, 0, currVal_13);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_3 = _co.headerText;
        _ck(_v, 12, 0, currVal_3);
        var currVal_9 = _co.left;
        var currVal_10 = 'left';
        var currVal_11 = _co.effectDuration;
        var currVal_12 = _co.easing;
        _ck(_v, 32, 0, currVal_9, currVal_10, currVal_11, currVal_12);
    });
}
exports.View_Carousel_0 = View_Carousel_0;
function View_Carousel_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'p-carousel', [], null, null, null, View_Carousel_0, exports.RenderType_Carousel)),
        i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(13811712, null, 1, i1.Carousel, [i0.ElementRef, i4.DomHandler, i0.Renderer2, i0.ChangeDetectorRef], null, null), i0.ɵqud(603979776, 1, { templates: 1 })], null, null);
}
exports.View_Carousel_Host_0 = View_Carousel_Host_0;
exports.CarouselNgFactory = i0.ɵccf('p-carousel', i1.Carousel, View_Carousel_Host_0, { numVisible: 'numVisible', firstVisible: 'firstVisible',
    headerText: 'headerText', circular: 'circular', breakpoint: 'breakpoint', responsive: 'responsive',
    autoplayInterval: 'autoplayInterval', effectDuration: 'effectDuration', easing: 'easing',
    pageLinks: 'pageLinks', style: 'style', styleClass: 'styleClass', value: 'value' }, { onPage: 'onPage' }, []);
