// primeng/components/common/shared.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/common/shared");
var i2 = require("@angular/common");
exports.SharedModuleNgFactory = i0.ɵcmf(i1.SharedModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.SharedModule, i1.SharedModule, [])]);
});
var styles_Header = [];
exports.RenderType_Header = i0.ɵcrt({ encapsulation: 2, styles: styles_Header,
    data: {} });
function View_Header_0(_l) {
    return i0.ɵvid(0, [i0.ɵncd(null, 0)], null, null);
}
exports.View_Header_0 = View_Header_0;
function View_Header_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-header', [], null, null, null, View_Header_0, exports.RenderType_Header)),
        i0.ɵdid(49152, null, 0, i1.Header, [], null, null)], null, null);
}
exports.View_Header_Host_0 = View_Header_Host_0;
exports.HeaderNgFactory = i0.ɵccf('p-header', i1.Header, View_Header_Host_0, {}, {}, ['*']);
var styles_Footer = [];
exports.RenderType_Footer = i0.ɵcrt({ encapsulation: 2, styles: styles_Footer,
    data: {} });
function View_Footer_0(_l) {
    return i0.ɵvid(0, [i0.ɵncd(null, 0)], null, null);
}
exports.View_Footer_0 = View_Footer_0;
function View_Footer_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-footer', [], null, null, null, View_Footer_0, exports.RenderType_Footer)),
        i0.ɵdid(49152, null, 0, i1.Footer, [], null, null)], null, null);
}
exports.View_Footer_Host_0 = View_Footer_Host_0;
exports.FooterNgFactory = i0.ɵccf('p-footer', i1.Footer, View_Footer_Host_0, {}, {}, ['*']);
var styles_Column = [];
exports.RenderType_Column = i0.ɵcrt({ encapsulation: 2, styles: styles_Column,
    data: {} });
function View_Column_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_Column_0 = View_Column_0;
function View_Column_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'p-column', [], null, null, null, View_Column_0, exports.RenderType_Column)),
        i0.ɵdid(1097728, null, 2, i1.Column, [], null, null),
        i0.ɵqud(603979776, 1, { templates: 1 }), i0.ɵqud(335544320, 2, { template: 0 })], null, null);
}
exports.View_Column_Host_0 = View_Column_Host_0;
exports.ColumnNgFactory = i0.ɵccf('p-column', i1.Column, View_Column_Host_0, { field: 'field', colId: 'colId', sortField: 'sortField', filterField: 'filterField',
    header: 'header', footer: 'footer', sortable: 'sortable', editable: 'editable', filter: 'filter',
    filterMatchMode: 'filterMatchMode', filterType: 'filterType', rowspan: 'rowspan',
    colspan: 'colspan', style: 'style', styleClass: 'styleClass', hidden: 'hidden', expander: 'expander',
    selectionMode: 'selectionMode', filterPlaceholder: 'filterPlaceholder', filterMaxlength: 'filterMaxlength',
    frozen: 'frozen' }, { sortFunction: 'sortFunction' }, []);
var styles_ColumnHeaderTemplateLoader = [];
exports.RenderType_ColumnHeaderTemplateLoader = i0.ɵcrt({ encapsulation: 2,
    styles: styles_ColumnHeaderTemplateLoader, data: {} });
function View_ColumnHeaderTemplateLoader_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_ColumnHeaderTemplateLoader_0 = View_ColumnHeaderTemplateLoader_0;
function View_ColumnHeaderTemplateLoader_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnHeaderTemplateLoader', [], null, null, null, View_ColumnHeaderTemplateLoader_0, exports.RenderType_ColumnHeaderTemplateLoader)), i0.ɵdid(245760, null, 0, i1.ColumnHeaderTemplateLoader, [i0.ViewContainerRef], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_ColumnHeaderTemplateLoader_Host_0 = View_ColumnHeaderTemplateLoader_Host_0;
exports.ColumnHeaderTemplateLoaderNgFactory = i0.ɵccf('p-columnHeaderTemplateLoader', i1.ColumnHeaderTemplateLoader, View_ColumnHeaderTemplateLoader_Host_0, { column: 'column' }, {}, []);
var styles_ColumnBodyTemplateLoader = [];
exports.RenderType_ColumnBodyTemplateLoader = i0.ɵcrt({ encapsulation: 2,
    styles: styles_ColumnBodyTemplateLoader, data: {} });
function View_ColumnBodyTemplateLoader_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_ColumnBodyTemplateLoader_0 = View_ColumnBodyTemplateLoader_0;
function View_ColumnBodyTemplateLoader_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnBodyTemplateLoader', [], null, null, null, View_ColumnBodyTemplateLoader_0, exports.RenderType_ColumnBodyTemplateLoader)), i0.ɵdid(770048, null, 0, i1.ColumnBodyTemplateLoader, [i0.ViewContainerRef], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_ColumnBodyTemplateLoader_Host_0 = View_ColumnBodyTemplateLoader_Host_0;
exports.ColumnBodyTemplateLoaderNgFactory = i0.ɵccf('p-columnBodyTemplateLoader', i1.ColumnBodyTemplateLoader, View_ColumnBodyTemplateLoader_Host_0, { column: 'column',
    rowData: 'rowData', rowIndex: 'rowIndex' }, {}, []);
var styles_ColumnFooterTemplateLoader = [];
exports.RenderType_ColumnFooterTemplateLoader = i0.ɵcrt({ encapsulation: 2,
    styles: styles_ColumnFooterTemplateLoader, data: {} });
function View_ColumnFooterTemplateLoader_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_ColumnFooterTemplateLoader_0 = View_ColumnFooterTemplateLoader_0;
function View_ColumnFooterTemplateLoader_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnFooterTemplateLoader', [], null, null, null, View_ColumnFooterTemplateLoader_0, exports.RenderType_ColumnFooterTemplateLoader)), i0.ɵdid(245760, null, 0, i1.ColumnFooterTemplateLoader, [i0.ViewContainerRef], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_ColumnFooterTemplateLoader_Host_0 = View_ColumnFooterTemplateLoader_Host_0;
exports.ColumnFooterTemplateLoaderNgFactory = i0.ɵccf('p-columnFooterTemplateLoader', i1.ColumnFooterTemplateLoader, View_ColumnFooterTemplateLoader_Host_0, { column: 'column' }, {}, []);
var styles_ColumnFilterTemplateLoader = [];
exports.RenderType_ColumnFilterTemplateLoader = i0.ɵcrt({ encapsulation: 2,
    styles: styles_ColumnFilterTemplateLoader, data: {} });
function View_ColumnFilterTemplateLoader_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_ColumnFilterTemplateLoader_0 = View_ColumnFilterTemplateLoader_0;
function View_ColumnFilterTemplateLoader_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnFilterTemplateLoader', [], null, null, null, View_ColumnFilterTemplateLoader_0, exports.RenderType_ColumnFilterTemplateLoader)), i0.ɵdid(245760, null, 0, i1.ColumnFilterTemplateLoader, [i0.ViewContainerRef], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_ColumnFilterTemplateLoader_Host_0 = View_ColumnFilterTemplateLoader_Host_0;
exports.ColumnFilterTemplateLoaderNgFactory = i0.ɵccf('p-columnFilterTemplateLoader', i1.ColumnFilterTemplateLoader, View_ColumnFilterTemplateLoader_Host_0, { column: 'column' }, {}, []);
var styles_TemplateLoader = [];
exports.RenderType_TemplateLoader = i0.ɵcrt({ encapsulation: 2,
    styles: styles_TemplateLoader, data: {} });
function View_TemplateLoader_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_TemplateLoader_0 = View_TemplateLoader_0;
function View_TemplateLoader_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-templateLoader', [], null, null, null, View_TemplateLoader_0, exports.RenderType_TemplateLoader)), i0.ɵdid(245760, null, 0, i1.TemplateLoader, [i0.ViewContainerRef], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_TemplateLoader_Host_0 = View_TemplateLoader_Host_0;
exports.TemplateLoaderNgFactory = i0.ɵccf('p-templateLoader', i1.TemplateLoader, View_TemplateLoader_Host_0, { template: 'template', data: 'data' }, {}, []);
var styles_Row = [];
exports.RenderType_Row = i0.ɵcrt({ encapsulation: 2, styles: styles_Row,
    data: {} });
function View_Row_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_Row_0 = View_Row_0;
function View_Row_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-row', [], null, null, null, View_Row_0, exports.RenderType_Row)), i0.ɵdid(49152, null, 1, i1.Row, [], null, null), i0.ɵqud(603979776, 1, { columns: 1 })], null, null);
}
exports.View_Row_Host_0 = View_Row_Host_0;
exports.RowNgFactory = i0.ɵccf('p-row', i1.Row, View_Row_Host_0, {}, {}, []);
var styles_HeaderColumnGroup = [];
exports.RenderType_HeaderColumnGroup = i0.ɵcrt({ encapsulation: 2,
    styles: styles_HeaderColumnGroup, data: {} });
function View_HeaderColumnGroup_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_HeaderColumnGroup_0 = View_HeaderColumnGroup_0;
function View_HeaderColumnGroup_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-headerColumnGroup', [], null, null, null, View_HeaderColumnGroup_0, exports.RenderType_HeaderColumnGroup)), i0.ɵdid(49152, null, 1, i1.HeaderColumnGroup, [], null, null), i0.ɵqud(603979776, 1, { rows: 1 })], null, null);
}
exports.View_HeaderColumnGroup_Host_0 = View_HeaderColumnGroup_Host_0;
exports.HeaderColumnGroupNgFactory = i0.ɵccf('p-headerColumnGroup', i1.HeaderColumnGroup, View_HeaderColumnGroup_Host_0, {}, {}, []);
var styles_FooterColumnGroup = [];
exports.RenderType_FooterColumnGroup = i0.ɵcrt({ encapsulation: 2,
    styles: styles_FooterColumnGroup, data: {} });
function View_FooterColumnGroup_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_FooterColumnGroup_0 = View_FooterColumnGroup_0;
function View_FooterColumnGroup_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-footerColumnGroup', [], null, null, null, View_FooterColumnGroup_0, exports.RenderType_FooterColumnGroup)), i0.ɵdid(49152, null, 1, i1.FooterColumnGroup, [], null, null), i0.ɵqud(603979776, 1, { rows: 1 })], null, null);
}
exports.View_FooterColumnGroup_Host_0 = View_FooterColumnGroup_Host_0;
exports.FooterColumnGroupNgFactory = i0.ɵccf('p-footerColumnGroup', i1.FooterColumnGroup, View_FooterColumnGroup_Host_0, {}, {}, []);
var styles_ColumnEditorTemplateLoader = [];
exports.RenderType_ColumnEditorTemplateLoader = i0.ɵcrt({ encapsulation: 2,
    styles: styles_ColumnEditorTemplateLoader, data: {} });
function View_ColumnEditorTemplateLoader_0(_l) {
    return i0.ɵvid(0, [], null, null);
}
exports.View_ColumnEditorTemplateLoader_0 = View_ColumnEditorTemplateLoader_0;
function View_ColumnEditorTemplateLoader_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 1, 'p-columnEditorTemplateLoader', [], null, null, null, View_ColumnEditorTemplateLoader_0, exports.RenderType_ColumnEditorTemplateLoader)), i0.ɵdid(245760, null, 0, i1.ColumnEditorTemplateLoader, [i0.ViewContainerRef], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_ColumnEditorTemplateLoader_Host_0 = View_ColumnEditorTemplateLoader_Host_0;
exports.ColumnEditorTemplateLoaderNgFactory = i0.ɵccf('p-columnEditorTemplateLoader', i1.ColumnEditorTemplateLoader, View_ColumnEditorTemplateLoader_Host_0, { column: 'column',
    rowData: 'rowData', rowIndex: 'rowIndex' }, {}, []);
