// @ng-bootstrap/ng-bootstrap/accordion/accordion.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@angular/common");
var i2 = require("@ng-bootstrap/ng-bootstrap/accordion/accordion");
var i3 = require("@ng-bootstrap/ng-bootstrap/accordion/accordion-config");
var styles_NgbAccordion = [];
exports.RenderType_NgbAccordion = i0.ɵcrt({ encapsulation: 2, styles: styles_NgbAccordion,
    data: {} });
function View_NgbAccordion_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_NgbAccordion_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_NgbAccordion_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'div', [['role',
                'tabpanel']], [[8, 'id', 0], [1, 'aria-labelledby', 0], [8, 'className', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n             '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbAccordion_4)),
        i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n        ']))], function (_ck, _v) {
        var currVal_3 = _v.parent.context.$implicit.contentTpl.templateRef;
        _ck(_v, 3, 0, currVal_3);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵinlineInterpolate(1, '', _v.parent.context.$implicit.id, '');
        var currVal_1 = (_v.parent.context.$implicit.id + '-header');
        var currVal_2 = i0.ɵinlineInterpolate(1, 'card-body ', (_v.parent.context.$implicit.isOpen ? 'show' : null), '');
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
    });
}
function View_NgbAccordion_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵeld(0, null, null, 13, 'div', [['class', 'card']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵeld(0, null, null, 7, 'div', [['role', 'tab']], [[8, 'id',
                0], [8, 'className', 0], [2, 'active', null]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n          '])),
        (_l()(), i0.ɵeld(0, null, null, 4, 'a', [['href', '']], [[2, 'text-muted',
                null], [1, 'tabindex', 0], [1, 'aria-expanded', 0], [1, 'aria-controls',
                0], [1, 'aria-disabled', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (!!_co.toggle(_v.context.$implicit.id) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            ',
            ''])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbAccordion_2)), i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0, 'ngTemplateOutlet'] }, null),
        (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_NgbAccordion_3)),
        i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_9 = ((_v.context.$implicit.titleTpl == null) ? null : _v.context.$implicit.titleTpl.templateRef);
        _ck(_v, 8, 0, currVal_9);
        var currVal_10 = (!_co.destroyOnHide || _v.context.$implicit.isOpen);
        _ck(_v, 13, 0, currVal_10);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = i0.ɵinlineInterpolate(1, '', _v.context.$implicit.id, '-header');
        var currVal_1 = ('card-header ' + (_v.context.$implicit.type ? ('card-' + _v.context.$implicit.type) : (_co.type ? ('card-' + _co.type) : '')));
        var currVal_2 = _v.context.$implicit.isOpen;
        _ck(_v, 3, 0, currVal_0, currVal_1, currVal_2);
        var currVal_3 = _v.context.$implicit.disabled;
        var currVal_4 = (_v.context.$implicit.disabled ? '-1' : null);
        var currVal_5 = _v.context.$implicit.isOpen;
        var currVal_6 = (_v.context.$implicit.isOpen ? _v.context.$implicit.id : null);
        var currVal_7 = _v.context.$implicit.disabled;
        _ck(_v, 5, 0, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7);
        var currVal_8 = _v.context.$implicit.title;
        _ck(_v, 6, 0, currVal_8);
    });
}
function View_NgbAccordion_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbAccordion_1)), i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.panels;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
exports.View_NgbAccordion_0 = View_NgbAccordion_0;
function View_NgbAccordion_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'ngb-accordion', [['role', 'tablist']], [[1, 'aria-multiselectable', 0]], null, null, View_NgbAccordion_0, exports.RenderType_NgbAccordion)), i0.ɵdid(2146304, null, 1, i2.NgbAccordion, [i3.NgbAccordionConfig], null, null), i0.ɵqud(603979776, 1, { panels: 1 })], null, function (_ck, _v) {
        var currVal_0 = !i0.ɵnov(_v, 1).closeOtherPanels;
        _ck(_v, 0, 0, currVal_0);
    });
}
exports.View_NgbAccordion_Host_0 = View_NgbAccordion_Host_0;
exports.NgbAccordionNgFactory = i0.ɵccf('ngb-accordion', i2.NgbAccordion, View_NgbAccordion_Host_0, { activeIds: 'activeIds', closeOtherPanels: 'closeOthers',
    destroyOnHide: 'destroyOnHide', type: 'type' }, { panelChange: 'panelChange' }, []);
