// primeng/components/fieldset/fieldset.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/fieldset/fieldset");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
exports.FieldsetModuleNgFactory = i0.ɵcmf(i1.FieldsetModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i1.FieldsetModule, i1.FieldsetModule, [])]);
});
var styles_Fieldset = [];
exports.RenderType_Fieldset = i0.ɵcrt({ encapsulation: 2, styles: styles_Fieldset,
    data: { 'animation': [{ type: 7, name: 'fieldsetContent', definitions: [{ type: 0, name: 'hidden',
                        styles: { type: 6, styles: { height: '0px' }, offset: null }, options: undefined },
                    { type: 0, name: 'visible', styles: { type: 6, styles: { height: '*' }, offset: null },
                        options: undefined }, { type: 1, expr: 'visible => hidden', animation: { type: 4,
                            styles: null, timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)' }, options: null },
                    { type: 1, expr: 'hidden => visible', animation: { type: 4, styles: null, timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)' },
                        options: null }], options: {} }] } });
function View_Fieldset_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class',
                'ui-fieldset-toggler fa fa-w']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'fa-minus': 0, 'fa-plus': 1 })], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-fieldset-toggler fa fa-w';
        var currVal_1 = _ck(_v, 2, 0, !_co.collapsed, _co.collapsed);
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_Fieldset_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 22, 'fieldset', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-fieldset ui-widget ui-widget-content ui-corner-all': 0,
            'ui-fieldset-toggleable': 1 }), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 6, 'legend', [['class', 'ui-fieldset-legend ui-corner-all ui-state-default ui-unselectable-text']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggle($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Fieldset_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ',
            '\n                '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 8, 'div', [['class', 'ui-fieldset-content-wrapper']], [[24, '@fieldsetContent', 0]], [[null, '@fieldsetContent.done']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('@fieldsetContent.done' === en)) {
                var pd_0 = (_co.onToggleDone($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-fieldset-content-wrapper-overflown': 0 }),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'ui-fieldset-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])), i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = _ck(_v, 3, 0, true, _co.toggleable);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 4, 0, currVal_2);
        var currVal_3 = _co.toggleable;
        _ck(_v, 9, 0, currVal_3);
        var currVal_6 = 'ui-fieldset-content-wrapper';
        var currVal_7 = _ck(_v, 16, 0, (_co.collapsed || _co.animating));
        _ck(_v, 15, 0, currVal_6, currVal_7);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_4 = _co.legend;
        _ck(_v, 10, 0, currVal_4);
        var currVal_5 = (_co.collapsed ? 'hidden' : 'visible');
        _ck(_v, 14, 0, currVal_5);
    });
}
exports.View_Fieldset_0 = View_Fieldset_0;
function View_Fieldset_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-fieldset', [], null, null, null, View_Fieldset_0, exports.RenderType_Fieldset)),
        i0.ɵdid(49152, null, 0, i1.Fieldset, [i0.ElementRef], null, null)], null, null);
}
exports.View_Fieldset_Host_0 = View_Fieldset_Host_0;
exports.FieldsetNgFactory = i0.ɵccf('p-fieldset', i1.Fieldset, View_Fieldset_Host_0, { legend: 'legend', toggleable: 'toggleable', collapsed: 'collapsed',
    style: 'style', styleClass: 'styleClass' }, { onBeforeToggle: 'onBeforeToggle', onAfterToggle: 'onAfterToggle' }, ['p-header', '*']);
