// primeng/components/contextmenu/contextmenu.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/contextmenu/contextmenu");
var i2 = require("@angular/common");
var i3 = require("@angular/router");
var i4 = require("primeng/components/dom/domhandler");
exports.ContextMenuModuleNgFactory = i0.ɵcmf(i1.ContextMenuModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.RouterModule, i3.RouterModule, [[2, i3.ɵa], [2, i3.Router]]), i0.ɵmpd(512, i1.ContextMenuModule, i1.ContextMenuModule, [])]);
});
var styles_ContextMenu = [];
exports.RenderType_ContextMenu = i0.ɵcrt({ encapsulation: 2, styles: styles_ContextMenu,
    data: {} });
function View_ContextMenu_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { containerViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, [[1, 0], ['container', 1]], null, 7, 'div', [], [[4, 'display', null]], null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers,
            i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 2, 'p-contextMenuSub', [['root', 'root']], null, null, null, View_ContextMenuSub_0, exports.RenderType_ContextMenuSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.ContextMenuSub, [i4.DomHandler,
            i1.ContextMenu], { item: [0, 'item'], root: [1, 'root'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _co.styleClass;
        var currVal_2 = 'ui-contextmenu ui-menu ui-widget ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-dynamic ui-shadow';
        _ck(_v, 3, 0, currVal_1, currVal_2);
        var currVal_3 = _co.style;
        _ck(_v, 4, 0, currVal_3);
        var currVal_4 = _co.model;
        var currVal_5 = 'root';
        _ck(_v, 8, 0, currVal_4, currVal_5);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.visible ? 'block' : 'none');
        _ck(_v, 2, 0, currVal_0);
    });
}
exports.View_ContextMenu_0 = View_ContextMenu_0;
function View_ContextMenu_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-contextMenu', [], null, null, null, View_ContextMenu_0, exports.RenderType_ContextMenu)),
        i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(4374528, null, 0, i1.ContextMenu, [i0.ElementRef, i4.DomHandler, i0.Renderer2], null, null)], null, null);
}
exports.View_ContextMenu_Host_0 = View_ContextMenu_Host_0;
exports.ContextMenuNgFactory = i0.ɵccf('p-contextMenu', i1.ContextMenu, View_ContextMenu_Host_0, { model: 'model', global: 'global', target: 'target',
    style: 'style', styleClass: 'styleClass', appendTo: 'appendTo' }, {}, []);
var styles_ContextMenuSub = [];
exports.RenderType_ContextMenuSub = i0.ɵcrt({ encapsulation: 2,
    styles: styles_ContextMenuSub, data: {} });
function View_ContextMenuSub_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'li', [['class',
                'ui-menu-separator ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                ']))], null, null);
}
function View_ContextMenuSub_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class',
                'ui-submenu-icon fa fa-fw fa-caret-right']], null, null, null, null, null))], null, null);
}
function View_ContextMenuSub_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _v.parent.parent.parent.context.$implicit.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_ContextMenuSub_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 13, 'a', [], [[8, 'href', 4], [1, 'target', 0], [1, 'title', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.itemClick($event, _v.parent.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-menuitem-link ui-corner-all': 0, 'ui-state-disabled': 1 }),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_5)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n                        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_6)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_3 = _v.parent.parent.context.$implicit.styleClass;
        var currVal_4 = _ck(_v, 2, 0, true, _v.parent.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_3, currVal_4);
        var currVal_5 = _v.parent.parent.context.$implicit.style;
        _ck(_v, 3, 0, currVal_5);
        var currVal_6 = _v.parent.parent.context.$implicit.items;
        _ck(_v, 6, 0, currVal_6);
        var currVal_7 = _v.parent.parent.context.$implicit.icon;
        _ck(_v, 9, 0, currVal_7);
    }, function (_ck, _v) {
        var currVal_0 = (_v.parent.parent.context.$implicit.url || '#');
        var currVal_1 = _v.parent.parent.context.$implicit.target;
        var currVal_2 = _v.parent.parent.context.$implicit.title;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_8 = _v.parent.parent.context.$implicit.label;
        _ck(_v, 12, 0, currVal_8);
    });
}
function View_ContextMenuSub_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'span', [['class',
                'ui-submenu-icon fa fa-fw fa-caret-right']], null, null, null, null, null))], null, null);
}
function View_ContextMenuSub_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-menuitem-icon fa fa-fw']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null)], function (_ck, _v) {
        var currVal_0 = 'ui-menuitem-icon fa fa-fw';
        var currVal_1 = _v.parent.parent.parent.context.$implicit.icon;
        _ck(_v, 1, 0, currVal_0, currVal_1);
    }, null);
}
function View_ContextMenuSub_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 18, 'a', [], [[1, 'target', 0], [1, 'title', 0], [1, 'target', 0], [8, 'href', 4]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 4).onClick($event.button, $event.ctrlKey, $event.metaKey, $event.shiftKey) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.itemClick($event, _v.parent.parent.context.$implicit) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-menuitem-link ui-corner-all': 0, 'ui-state-disabled': 1 }),
        i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), i0.ɵdid(671744, [[2, 4]], 0, i3.RouterLinkWithHref, [i3.Router, i3.ActivatedRoute, i2.LocationStrategy], { routerLink: [0, 'routerLink'] }, null), i0.ɵdid(1720320, null, 2, i3.RouterLinkActive, [i3.Router, i0.ElementRef, i0.Renderer2, i0.ChangeDetectorRef], { routerLinkActiveOptions: [0, 'routerLinkActiveOptions'], routerLinkActive: [1,
                'routerLinkActive'] }, null), i0.ɵqud(603979776, 1, { links: 1 }), i0.ɵqud(603979776, 2, { linksWithHrefs: 1 }), i0.ɵpod({ exact: 0 }), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_8)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_9)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'ui-menuitem-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', ''])), (_l()(), i0.ɵted(null, ['\n                    ']))], function (_ck, _v) {
        var currVal_4 = _v.parent.parent.context.$implicit.styleClass;
        var currVal_5 = _ck(_v, 2, 0, true, _v.parent.parent.context.$implicit.disabled);
        _ck(_v, 1, 0, currVal_4, currVal_5);
        var currVal_6 = _v.parent.parent.context.$implicit.style;
        _ck(_v, 3, 0, currVal_6);
        var currVal_7 = _v.parent.parent.context.$implicit.routerLink;
        _ck(_v, 4, 0, currVal_7);
        var currVal_8 = (_v.parent.parent.context.$implicit.routerLinkActiveOptions || _ck(_v, 8, 0, false));
        var currVal_9 = 'ui-state-active';
        _ck(_v, 5, 0, currVal_8, currVal_9);
        var currVal_10 = _v.parent.parent.context.$implicit.items;
        _ck(_v, 11, 0, currVal_10);
        var currVal_11 = _v.parent.parent.context.$implicit.icon;
        _ck(_v, 14, 0, currVal_11);
    }, function (_ck, _v) {
        var currVal_0 = _v.parent.parent.context.$implicit.target;
        var currVal_1 = _v.parent.parent.context.$implicit.title;
        var currVal_2 = i0.ɵnov(_v, 4).target;
        var currVal_3 = i0.ɵnov(_v, 4).href;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
        var currVal_12 = _v.parent.parent.context.$implicit.label;
        _ck(_v, 17, 0, currVal_12);
    });
}
function View_ContextMenuSub_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-contextMenuSub', [['class', 'ui-submenu']], null, null, null, View_ContextMenuSub_0, exports.RenderType_ContextMenuSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.ContextMenuSub, [i4.DomHandler,
            i1.ContextMenu], { item: [0, 'item'] }, null)], function (_ck, _v) {
        var currVal_0 = _v.parent.parent.context.$implicit;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
function View_ContextMenuSub_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, [['item', 1]], null, 12, 'li', [], [[4, 'display', null]], [[null, 'mouseenter'], [null, 'mouseleave']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('mouseenter' === en)) {
                var pd_0 = (_co.onItemMouseEnter($event, i0.ɵnov(_v, 0), _v.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            if (('mouseleave' === en)) {
                var pd_1 = (_co.onItemMouseLeave($event, i0.ɵnov(_v, 0)) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0, 'ngClass'] }, null),
        i0.ɵpod({ 'ui-menuitem ui-widget ui-corner-all': 0, 'ui-menu-parent': 1, 'ui-menuitem-active': 2 }),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_4)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_7)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_10)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _ck(_v, 2, 0, true, _v.parent.context.$implicit.items, (i0.ɵnov(_v, 0) == _co.activeItem));
        _ck(_v, 1, 0, currVal_1);
        var currVal_2 = !_v.parent.context.$implicit.routerLink;
        _ck(_v, 5, 0, currVal_2);
        var currVal_3 = _v.parent.context.$implicit.routerLink;
        _ck(_v, 8, 0, currVal_3);
        var currVal_4 = _v.parent.context.$implicit.items;
        _ck(_v, 11, 0, currVal_4);
    }, function (_ck, _v) {
        var currVal_0 = ((_v.parent.context.$implicit.visible === false) ? 'none' : 'block');
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_ContextMenuSub_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n                '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_3)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.separator;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.separator;
        _ck(_v, 4, 0, currVal_1);
    }, null);
}
function View_ContextMenuSub_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'ul', [['class', 'ui-menu-list']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.listClick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-helper-reset': 0, 'ui-widget-content ui-corner-all ui-helper-clearfix ui-menu-child ui-shadow': 1 }),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ContextMenuSub_1)), i0.ɵdid(802816, null, 0, i2.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'ui-menu-list';
        var currVal_1 = _ck(_v, 3, 0, _co.root, !_co.root);
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = (_co.root ? _co.item : _co.item.items);
        _ck(_v, 6, 0, currVal_2);
    }, null);
}
exports.View_ContextMenuSub_0 = View_ContextMenuSub_0;
function View_ContextMenuSub_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-contextMenuSub', [], null, null, null, View_ContextMenuSub_0, exports.RenderType_ContextMenuSub)), i0.ɵprd(512, null, i4.DomHandler, i4.DomHandler, []), i0.ɵdid(49152, null, 0, i1.ContextMenuSub, [i4.DomHandler,
            i1.ContextMenu], null, null)], null, null);
}
exports.View_ContextMenuSub_Host_0 = View_ContextMenuSub_Host_0;
exports.ContextMenuSubNgFactory = i0.ɵccf('p-contextMenuSub', i1.ContextMenuSub, View_ContextMenuSub_Host_0, { item: 'item', root: 'root' }, {}, []);
