// primeng/components/progressbar/progressbar.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/progressbar/progressbar");
var i2 = require("@angular/common");
exports.ProgressBarModuleNgFactory = i0.ɵcmf(i1.ProgressBarModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.ProgressBarModule, i1.ProgressBarModule, [])]);
});
var styles_ProgressBar = [];
exports.RenderType_ProgressBar = i0.ɵcrt({ encapsulation: 2, styles: styles_ProgressBar,
    data: {} });
function View_ProgressBar_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['class',
                'ui-progressbar-label']], [[4, 'display', null]], null, null, null, null)), (_l()(), i0.ɵted(null, ['', '', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.value ? 'block' : 'none');
        _ck(_v, 0, 0, currVal_0);
        var currVal_1 = _co.value;
        var currVal_2 = _co.unit;
        _ck(_v, 1, 0, currVal_1, currVal_2);
    });
}
function View_ProgressBar_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 6, 'div', [['aria-valuemax', '100'], ['aria-valuemin',
                '0'], ['class', 'ui-progressbar ui-widget ui-widget-content ui-corner-all'],
            ['role', 'progressbar']], [[1, 'aria-valuenow', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'div', [['class', 'ui-progressbar-value ui-progressbar-value-animate ui-widget-header ui-corner-all'],
            ['style', 'display:block']], [[4, 'width', null]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ProgressBar_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = _co.showValue;
        _ck(_v, 6, 0, currVal_2);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.value;
        _ck(_v, 1, 0, currVal_0);
        var currVal_1 = (_co.value + '%');
        _ck(_v, 3, 0, currVal_1);
    });
}
exports.View_ProgressBar_0 = View_ProgressBar_0;
function View_ProgressBar_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-progressBar', [], null, null, null, View_ProgressBar_0, exports.RenderType_ProgressBar)),
        i0.ɵdid(49152, null, 0, i1.ProgressBar, [], null, null)], null, null);
}
exports.View_ProgressBar_Host_0 = View_ProgressBar_Host_0;
exports.ProgressBarNgFactory = i0.ɵccf('p-progressBar', i1.ProgressBar, View_ProgressBar_Host_0, { value: 'value', showValue: 'showValue', unit: 'unit' }, {}, []);
