// primeng/components/panel/panel.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/panel/panel");
var i2 = require("@angular/common");
var i3 = require("primeng/components/common/shared");
exports.PanelModuleNgFactory = i0.ɵcmf(i1.PanelModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i3.SharedModule, i3.SharedModule, []), i0.ɵmpd(512, i1.PanelModule, i1.PanelModule, [])]);
});
var styles_Panel = [];
exports.RenderType_Panel = i0.ɵcrt({ encapsulation: 2, styles: styles_Panel,
    data: { 'animation': [{ type: 7, name: 'panelContent', definitions: [{ type: 0, name: 'hidden',
                        styles: { type: 6, styles: { height: '0' }, offset: null }, options: undefined },
                    { type: 0, name: 'visible', styles: { type: 6, styles: { height: '*' }, offset: null },
                        options: undefined }, { type: 1, expr: 'visible <=> hidden', animation: { type: 4,
                            styles: null, timings: '400ms cubic-bezier(0.86, 0, 0.07, 1)' }, options: null }],
                options: {} }] } });
function View_Panel_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class',
                'ui-panel-title']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.header;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_Panel_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'a', [['class',
                'ui-panel-titlebar-icon ui-panel-titlebar-toggler ui-corner-all ui-state-default'],
            ['href', '#']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggle($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [], [[8, 'className',
                0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.collapsed ? ('fa fa-fw ' + _co.expandIcon) : ('fa fa-fw ' + _co.collapseIcon));
        _ck(_v, 2, 0, currVal_0);
    });
}
function View_Panel_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'ui-panel-footer ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        i0.ɵncd(null, 2), (_l()(), i0.ɵted(null, ['\n                ']))], null, null);
}
function View_Panel_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 27, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 9, 'div', [['class', 'ui-panel-titlebar ui-widget-header ui-helper-clearfix ui-corner-all']], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Panel_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n                '])), i0.ɵncd(null, 0), (_l()(),
            i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Panel_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 11, 'div', [['class', 'ui-panel-content-wrapper']], [[24, '@panelContent', 0]], [[null, '@panelContent.done']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('@panelContent.done' === en)) {
                var pd_0 = (_co.onToggleDone($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'ui-panel-content-wrapper-overflown': 0 }),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'ui-panel-content ui-widget-content']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])), i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵted(null, ['\n                \n                '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Panel_3)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-panel ui-widget ui-widget-content ui-corner-all';
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 3, 0, currVal_2);
        var currVal_3 = _co.header;
        _ck(_v, 8, 0, currVal_3);
        var currVal_4 = _co.toggleable;
        _ck(_v, 13, 0, currVal_4);
        var currVal_6 = 'ui-panel-content-wrapper';
        var currVal_7 = _ck(_v, 18, 0, (_co.collapsed || _co.animating));
        _ck(_v, 17, 0, currVal_6, currVal_7);
        var currVal_8 = _co.footerFacet;
        _ck(_v, 26, 0, currVal_8);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_5 = (_co.collapsed ? 'hidden' : 'visible');
        _ck(_v, 16, 0, currVal_5);
    });
}
exports.View_Panel_0 = View_Panel_0;
function View_Panel_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-panel', [], null, null, null, View_Panel_0, exports.RenderType_Panel)), i0.ɵdid(49152, null, 1, i1.Panel, [i0.ElementRef], null, null), i0.ɵqud(335544320, 1, { footerFacet: 0 })], null, null);
}
exports.View_Panel_Host_0 = View_Panel_Host_0;
exports.PanelNgFactory = i0.ɵccf('p-panel', i1.Panel, View_Panel_Host_0, { toggleable: 'toggleable', header: 'header', collapsed: 'collapsed',
    style: 'style', styleClass: 'styleClass', expandIcon: 'expandIcon', collapseIcon: 'collapseIcon' }, { collapsedChange: 'collapsedChange', onBeforeToggle: 'onBeforeToggle', onAfterToggle: 'onAfterToggle' }, ['p-header', '*', 'p-footer']);
