// @ng-bootstrap/ng-bootstrap/progressbar/progressbar.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@angular/common");
var i2 = require("@ng-bootstrap/ng-bootstrap/progressbar/progressbar");
var i3 = require("@ng-bootstrap/ng-bootstrap/progressbar/progressbar-config");
var styles_NgbProgressbar = [];
exports.RenderType_NgbProgressbar = i0.ɵcrt({ encapsulation: 2,
    styles: styles_NgbProgressbar, data: {} });
function View_NgbProgressbar_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'span', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['', '%']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.getPercentValue();
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_NgbProgressbar_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵeld(0, null, null, 8, 'div', [['class', 'progress']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(),
            i0.ɵeld(0, null, null, 5, 'div', [['aria-valuemin', '0'], ['role',
                    'progressbar']], [[8, 'className', 0], [4, 'width', '%'], [1, 'aria-valuenow', 0],
                [1, 'aria-valuemax', 0]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_NgbProgressbar_1)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(),
            i0.ɵted(null, ['\n    '])), (_l()(), i0.ɵted(null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_4 = _co.showValue;
        _ck(_v, 6, 0, currVal_4);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = i0.ɵinlineInterpolate(3, 'progress-bar', (_co.type ? (' bg-' + _co.type) : ''), '', (_co.animated ? ' progress-bar-animated' : ''), '', (_co.striped ? ' progress-bar-striped' : ''), '');
        var currVal_1 = _co.getPercentValue();
        var currVal_2 = _co.getValue();
        var currVal_3 = _co.max;
        _ck(_v, 3, 0, currVal_0, currVal_1, currVal_2, currVal_3);
    });
}
exports.View_NgbProgressbar_0 = View_NgbProgressbar_0;
function View_NgbProgressbar_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'ngb-progressbar', [], null, null, null, View_NgbProgressbar_0, exports.RenderType_NgbProgressbar)), i0.ɵdid(49152, null, 0, i2.NgbProgressbar, [i3.NgbProgressbarConfig], null, null)], null, null);
}
exports.View_NgbProgressbar_Host_0 = View_NgbProgressbar_Host_0;
exports.NgbProgressbarNgFactory = i0.ɵccf('ngb-progressbar', i2.NgbProgressbar, View_NgbProgressbar_Host_0, { max: 'max', animated: 'animated', striped: 'striped',
    showValue: 'showValue', type: 'type', value: 'value' }, {}, ['*']);
