// primeng/components/radiobutton/radiobutton.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/radiobutton/radiobutton");
var i2 = require("@angular/common");
var i3 = require("@angular/forms");
exports.RadioButtonModuleNgFactory = i0.ɵcmf(i1.RadioButtonModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.RadioButtonModule, i1.RadioButtonModule, [])]);
});
var styles_RadioButton = [];
exports.RenderType_RadioButton = i0.ɵcrt({ encapsulation: 2, styles: styles_RadioButton,
    data: {} });
function View_RadioButton_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'label', [['class',
                'ui-radiobutton-label']], [[1, 'for', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.select() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod({ 'ui-label-active': 0, 'ui-label-disabled': 1,
            'ui-label-focus': 2 }), (_l()(), i0.ɵted(null, ['', '']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = 'ui-radiobutton-label';
        var currVal_2 = _ck(_v, 2, 0, i0.ɵnov(_v.parent, 8).checked, _co.disabled, _co.focused);
        _ck(_v, 1, 0, currVal_1, currVal_2);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.inputId;
        _ck(_v, 0, 0, currVal_0);
        var currVal_3 = _co.label;
        _ck(_v, 3, 0, currVal_3);
    });
}
function View_RadioButton_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { inputViewChild: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 17, 'div', [], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵdid(278528, null, 0, i2.NgStyle, [i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngStyle: [0,
                'ngStyle'] }, null), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'ui-helper-hidden-accessible']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, [[1,
                0], ['rb', 1]], null, 0, 'input', [['type', 'radio']], [[1, 'id', 0], [1, 'name',
                0], [1, 'value', 0], [1, 'tabindex', 0], [8, 'checked', 0]], [[null, 'change'],
            [null, 'focus'], [null, 'blur']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('change' === en)) {
                var pd_0 = (_co.onChange($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('focus' === en)) {
                var pd_1 = (_co.onFocus($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('blur' === en)) {
                var pd_2 = (_co.onBlur($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 7, 'div', [], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.handleClick() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { ngClass: [0,
                'ngClass'] }, null), i0.ɵpod({ 'ui-radiobutton-box ui-widget ui-state-default': 0,
            'ui-state-active': 1, 'ui-state-disabled': 2, 'ui-state-focus': 3 }), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 2, 'span', [['class', 'ui-radiobutton-icon ui-clickable']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod({ 'fa fa-circle': 0 }),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_RadioButton_1)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.styleClass;
        var currVal_1 = 'ui-radiobutton ui-widget';
        _ck(_v, 3, 0, currVal_0, currVal_1);
        var currVal_2 = _co.style;
        _ck(_v, 4, 0, currVal_2);
        var currVal_8 = _ck(_v, 13, 0, true, i0.ɵnov(_v, 8).checked, _co.disabled, _co.focused);
        _ck(_v, 12, 0, currVal_8);
        var currVal_9 = 'ui-radiobutton-icon ui-clickable';
        var currVal_10 = _ck(_v, 17, 0, i0.ɵnov(_v, 8).checked);
        _ck(_v, 16, 0, currVal_9, currVal_10);
        var currVal_11 = _co.label;
        _ck(_v, 22, 0, currVal_11);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_3 = _co.inputId;
        var currVal_4 = _co.name;
        var currVal_5 = _co.value;
        var currVal_6 = _co.tabindex;
        var currVal_7 = _co.checked;
        _ck(_v, 8, 0, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7);
    });
}
exports.View_RadioButton_0 = View_RadioButton_0;
function View_RadioButton_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'p-radioButton', [], null, null, null, View_RadioButton_0, exports.RenderType_RadioButton)),
        i0.ɵprd(5120, null, i3.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i1.RadioButton]), i0.ɵdid(49152, null, 0, i1.RadioButton, [i0.ChangeDetectorRef], null, null)], null, null);
}
exports.View_RadioButton_Host_0 = View_RadioButton_Host_0;
exports.RadioButtonNgFactory = i0.ɵccf('p-radioButton', i1.RadioButton, View_RadioButton_Host_0, { value: 'value', name: 'name', disabled: 'disabled',
    label: 'label', tabindex: 'tabindex', inputId: 'inputId', style: 'style', styleClass: 'styleClass' }, { onClick: 'onClick' }, []);
