// primeng/components/chart/chart.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("primeng/components/chart/chart");
var i2 = require("@angular/common");
exports.ChartModuleNgFactory = i0.ɵcmf(i1.ChartModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.NgLocalization, i2.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(512, i2.CommonModule, i2.CommonModule, []), i0.ɵmpd(512, i1.ChartModule, i1.ChartModule, [])]);
});
var styles_UIChart = [];
exports.RenderType_UIChart = i0.ɵcrt({ encapsulation: 2, styles: styles_UIChart,
    data: {} });
function View_UIChart_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'canvas', [], [[1, 'width',
                0], [1, 'height', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onCanvasClick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.width;
        var currVal_1 = _co.height;
        _ck(_v, 3, 0, currVal_0, currVal_1);
    });
}
exports.View_UIChart_0 = View_UIChart_0;
function View_UIChart_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'p-chart', [], null, null, null, View_UIChart_0, exports.RenderType_UIChart)),
        i0.ɵdid(4374528, null, 0, i1.UIChart, [i0.ElementRef], null, null)], null, null);
}
exports.View_UIChart_Host_0 = View_UIChart_Host_0;
exports.UIChartNgFactory = i0.ɵccf('p-chart', i1.UIChart, View_UIChart_Host_0, { type: 'type', options: 'options', width: 'width', height: 'height',
    data: 'data' }, { onDataSelect: 'onDataSelect' }, []);
